## Author: Lucy Rogers     
## Ecological Genomics: Spring 2018

### Overall Description of notebook      

This notebook will catalog my entries throughout the semester, including all my code, results, and interpretations


### Date started: (2018-01-24)   
### Date end:   (year-month-day)    

### Philosophy   
Science should be reproducible and one of the best ways to achieve this is by logging research activities in a notebook. Because science/biology has increasingly become computational, it is easier to document computational projects in an electronic form, which can be shared online through Github.    

### Helpful features of the notebook     

**It is absolutely critical for your future self and others to follow your work.**     

* The notebook is set up with a series of internal links from the table of contents.    
* All notebooks should have a table of contents which has the "Page", date, and title (information that allows the reader to understand your work).     
* Also, one of the perks of keeping all activities in a single document is that you can **search and find elements quickly**.     
* Lastly, you can share specific entries because of the three "#" automatically creates a link when the notebook renders on github.      


<a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.  


### Table of contents for 60 entries (Format is *Page: Date(with year-month-day). Title*)        
* [Page 1: 2018-01-24](#id-section1). Intro to Github, RMarkdown, and UNIX command-line
* [Page 2: 2018-01-26](#id-section2). Independent practice with UNIX
* [Page 3: 2018-01-29](#id-section3). Visualize data by creating Fast QC report
* [Page 4:2018-01-31](#id-section4). Clean data using trimmomatic, visualize again, create sam directory and align sequences
* [Page 5:2018-02-05](#id-section5). Explore sam output, install DESeq into RStudio
* [Page 6:2018-02-07](#id-section6). Review bash script, use DESeq in R to write code to test for differences in gene expression
* [Page 7:2018-02-12](#id-section7). Writing R Script to test for differential gene expression
* [Page 8:2018-02-14](#id-section8). Writing R Script, actual script is posted here
* [Page 9:2018-02-21](#id-section9). Writing R Script
* [Page 10: 2018-02-26](#id-section10).Intro to SNP and Genotype Calling
* [Page 11:](#id-section11).
* [Page 12:](#id-section12).
* [Page 13:](#id-section13).
* [Page 14:](#id-section14).
* [Page 15:](#id-section15).
* [Page 16:](#id-section16).
* [Page 17:](#id-section17).
* [Page 18:](#id-section18).
* [Page 19:](#id-section19).
* [Page 20:](#id-section20).
* [Page 21:](#id-section21).
* [Page 22:](#id-section22).
* [Page 23:](#id-section23).
* [Page 24:](#id-section24).
* [Page 25:](#id-section25).
* [Page 26:](#id-section26).
* [Page 27:](#id-section27).
* [Page 28:](#id-section28).
* [Page 29:](#id-section29).
* [Page 30:](#id-section30).
* [Page 31:](#id-section31).
* [Page 32:](#id-section32).
* [Page 33:](#id-section33).
* [Page 34:](#id-section34).
* [Page 35:](#id-section35).
* [Page 36:](#id-section36).
* [Page 37:](#id-section37).
* [Page 38:](#id-section38).
* [Page 39:](#id-section39).
* [Page 40:](#id-section40).
* [Page 41:](#id-section41).
* [Page 42:](#id-section42).
* [Page 43:](#id-section43).
* [Page 44:](#id-section44).
* [Page 45:](#id-section45).
* [Page 46:](#id-section46).
* [Page 47:](#id-section47).
* [Page 48:](#id-section48).
* [Page 49:](#id-section49).
* [Page 50:](#id-section50).
* [Page 51:](#id-section51).
* [Page 52:](#id-section52).
* [Page 53:](#id-section53).
* [Page 54:](#id-section54).
* [Page 55:](#id-section55).
* [Page 56:](#id-section56).
* [Page 57:](#id-section57).
* [Page 58:](#id-section58).
* [Page 59:](#id-section59).
* [Page 60:](#id-section60).

------
<div id='id-section1'/>
### Page 1: 2018-01-24. Notes on using Github, Rmarkdown, and the UNIX command-line

Today, we created our Github repos for the course and began our notebooks.

Other goals for today:

* publish our notbooks to github
* log into the UNIX server


------
<div id='id-section2'/>
### Page 2: 2018-01-28 

Today, I reviewed the UNIX commands we learned on 2018-01-24. I reviewed the tutorial "Intro to connecting to unix servers and navigating the command-line" and changed the setting such that terminal will check with me before deleting a file via the rm function. Tomorrow in class, I will ask about why I cannot see my name linked under the class member page link on the course github page.

The most useful UNIX commands were

pwd = present working directory
mkdir = make directory
ll = long list
cd = change directory
cp filename path_to_destination = copy
head filename = print head...can also do with tail
grep 'word' filename = pull out all entries in file with the 'word' in them
grep 'IT' cols_data.txt >cols_ITonly.txt = save the ITonly output in the new file
man command = go to manual and tell me more about this command
wc = tally up lines, words, and characters in a dataset
rm filename = remove filename
grep 'F' filename | wc = use pipe to apply 2nd command to 1st command


------
<div id='id-section3'/>
### Page 3: 2018-01-29



Today, we learned about the bull headed beetle dataset and about the RNASeq data processing 'pipeline'.

First, we found our specific file within the group data file and used the zcat command to open it and the pipe command | to only show us the first 4 lines. We learned how to interpret this output data:
1. after @ is information about the reads
2. Then the actual sequence
3. sometimes more info
4. Quality data, the key is here http://www.drive5.com/usearch/manual/quality_score.html.


[lerogers@pbio381 ~]$ cd /data/project_data/beetles/rawdata/
[lerogers@pbio381 rawdata]$ zcat WA_PP1_F3_ATGTCA_L003_R1_001.fastq.gz  | head -n4
@CCRI0219:155:C2LNBACXX:3:1101:1226:1976 1:N:0:ATGTCA
TGGCGCCAAAAATAAGATGGAGAAAAAGAAATTAGTTGCGCATGCTGATGTTAAGTCTCGAGACCGGTACTATTTTAAATAAAAAAAATTTTATTTGATG
+
B@CFFFFFDDFFHIFGGGHGFHGIIIGIGHIIIJJDHIGGIGDHHGDCGIIEGE@=DHGIGGGHFDD>>?CDDEEDDDDEEDDCDDDDDCDDDDDDEDDC


Then we used the fastqc command to visualize our data and copied it into a new directory we made called fastqcoutput: 

[lerogers@pbio381 rawdata]$ cd ~
[lerogers@pbio381 ~]$ mkdir fastqcoutput
[lerogers@pbio381 ~]$ fastqc /data/project_data/beetles/rawdata/WA_PP1_F3_ATGTCA_L003_R1_001.fastq.gz -o ~/fastqcoutput
Started analysis of WA_PP1_F3_ATGTCA_L003_R1_001.fastq.gz
Approx 5% complete for WA_PP1_F3_ATGTCA_L003_R1_001.fastq.gz
...
Analysis complete for WA_PP1_F3_ATGTCA_L003_R1_001.fastq.gz


Using a second terminal window, we copied the fastqc output from the fastqcoutput folder on our home directory to a folder we created on our computers; mine is in the folder 'terminal output files' on my desktop. I did cd drag_file_symbol_here

scp = secure copy
 . = put it here

Terminal window 2:
ip0af52b37:~ Lucy$ cd /Users/Lucy/Desktop/Ecological\ Genomics\ Terminal\ Output\ files 
ip0af52b37:Ecological Genomics Terminal Output files Lucy$ pwd
/Users/Lucy/Desktop/Ecological Genomics Terminal Output files
ip0af52b37:Ecological Genomics Terminal Output files Lucy$ scp lerogers@pbio381.uvm.edu:~/fastqcoutput/* .
lerogers@pbio381.uvm.edu's password: 
WA_PP1_F3_ATGTCA_L003_R1_001_fastqc.html      100%  331KB 331.1KB/s   00:00    
WA_PP1_F3_ATGTCA_L003_R1_001_fastqc.zip       100%  428KB 428.0KB/s   00:00    
ip0af52b37:Ecological Genomics Terminal Output files Lucy$ 



------
<div id='id-section4'/>
### Page 4: 2018-01-31

We made a cleanreads folder on our home directory and then copied the example bash script from the group data folder to the myscripts folder on our home directory. We then opened the script using vim, which brought us to a new window on terminal. 


[lerogers@pbio381 ~]$ cd ~/
[lerogers@pbio381 ~]$ mkdir cleanreads
[lerogers@pbio381 ~]$ ll
total 4
drwxr-xr-x. 2 lerogers users    6 Jan 31 10:21 cleanreads
drwxr-xr-x. 2 lerogers users 4096 Jan 29 11:07 fastqcoutput
drwxr-xr-x. 2 lerogers users   64 Jan 24 11:30 mydata
drwxr-xr-x. 2 lerogers users    6 Jan 24 11:18 myscripts
[lerogers@pbio381 myscripts]$ cp /data/scripts/trim_example.sh ~/myscripts/
[lerogers@pbio381 myscripts]$ ll
total 4
-rwxr--r--. 1 lerogers users 780 Jan 31 10:27 trim_example.sh
[lerogers@pbio381 myscripts]$ vim trim_example.sh
[lerogers@pbio381 myscripts]$ ll
total 4
-rwxr--r--. 1 lerogers users 764 Jan 31 10:43 trim_example.sh
[lerogers@pbio381 myscripts]$ vim trim_example.sh
[lerogers@pbio381 myscripts]$ ll
total 8
-rwxr--r--. 1 lerogers users 764 Jan 31 10:43 trim_example.sh
-rw-r--r--. 1 lerogers users 764 Jan 31 10:48 trim_F3
[lerogers@pbio381 myscripts]$ ll
total 8
-rwxr--r--. 1 lerogers users 764 Jan 31 10:43 trim_example.sh
-rw-r--r--. 1 lerogers users 764 Jan 31 10:48 trim_F3
[lerogers@pbio381 myscripts]$ vim trim_example.sh
[lerogers@pbio381 myscripts]$ ll
total 12
-rwxr--r--. 1 lerogers users 764 Jan 31 10:43 trim_example.sh
-rw-r--r--. 1 lerogers users 764 Jan 31 10:48 trim_F3
-rw-r--r--. 1 lerogers users 764 Jan 31 10:49 trim_F3.sh
[lerogers@pbio381 myscripts]$ rm trim_F3
rm: remove regular file ‘trim_F3’? yes
[lerogers@pbio381 myscripts]$ ll
total 8
-rwxr--r--. 1 lerogers users 764 Jan 31 10:43 trim_example.sh
-rw-r--r--. 1 lerogers users 764 Jan 31 10:49 trim_F3.sh



Now, we ran the trimmomatic script by first using chmod, which allowed us to change the permissions so that the 'user' could 'execute' the script, then ran the command ./trim_F3.sh which told it to run. 

[lerogers@pbio381 myscripts]$ chmod u+x trim_F3.sh
[lerogers@pbio381 myscripts]$ ll
total 8
-rwxr--r--. 1 lerogers users 764 Jan 31 10:43 trim_example.sh
-rwxr--r--. 1 lerogers users 764 Jan 31 10:49 trim_F3.sh
[lerogers@pbio381 myscripts]$ vim trim_f3.sh
[lerogers@pbio381 myscripts]$ vim trim_F3.sh
[lerogers@pbio381 myscripts]$ rm trim_example.sh
rm: remove regular file ‘trim_example.sh’? yes
[lerogers@pbio381 myscripts]$ ll
total 4
-rwxr--r--. 1 lerogers users 764 Jan 31 10:49 trim_F3.sh
[lerogers@pbio381 myscripts]$ ./trim_F3.sh
TrimmomaticPE: Started with arguments: -threads 1 -phred33 /data/project_data/beetles/rawdata/WA_PP1_F3_ATGTCA_L003_R1_001.fastq.gz /data/project_data/beetles/rawdata/WA_PP1_F3_ATGTCA_L003_R2_001.fastq.gz /users/l/e/lerogers/cleanreads/WA_PP1_F3_R1_clean_paired.fa /users/l/e/lerogers/cleanreads/WA_PP1_F3_R1_clean_unpaired.fa /users/l/e/lerogers/cleanreads/WA_PP1_F3_R2_clean_paired.fa /users/l/e/lerogers/cleanreads/WA_PP1_F3_R2_clean_unpaired.fa ILLUMINACLIP:/data/popgen/Trimmomatic-0.33/adapters/TruSeq3-PE.fa:2:30:10 LEADING:28 TRAILING:28 SLIDINGWINDOW:6:28 HEADCROP:12 MINLEN:35
Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'
ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
Input Read Pairs: 16000000 Both Surviving: 11356978 (70.98%) Forward Only Surviving: 1897057 (11.86%) Reverse Only Surviving: 800823 (5.01%) Dropped: 1945142 (12.16%)
TrimmomaticPE: Completed successfully
[lerogers@pbio381 myscripts]$ cd ~/cleanreads/
[lerogers@pbio381 cleanreads]$ ll
total 5548308
-rw-r--r--. 1 lerogers users 2590374644 Jan 31 10:59 WA_PP1_F3_R1_clean_paired.fa
-rw-r--r--. 1 lerogers users  369021891 Jan 31 10:59 WA_PP1_F3_R1_clean_unpaired.fa
-rw-r--r--. 1 lerogers users 2562494874 Jan 31 10:59 WA_PP1_F3_R2_clean_paired.fa
-rw-r--r--. 1 lerogers users  159565225 Jan 31 10:59 WA_PP1_F3_R2_clean_unpaired.fa


Now, finally, we ran fastqc again to visualize after trimming, placing the final output in our fastqc file: 

[lerogers@pbio381 cleanreads]$ fastqc ~/cleanreads/WA_PP1_F3_R1_clean_paired.fa -o ~/fastqcoutput/
Started analysis of WA_PP1_F3_R1_clean_paired.fa
Approx 5% complete for WA_PP1_F3_R1_clean_paired.fa
...
Analysis complete for WA_PP1_F3_R1_clean_paired.fa
[lerogers@pbio381 cleanreads]$ cd ~/
[lerogers@pbio381 ~]$ ll
total 8
drwxr-xr-x. 2 lerogers users 4096 Jan 31 10:55 cleanreads
drwxr-xr-x. 2 lerogers users 4096 Jan 31 11:10 fastqcoutput
drwxr-xr-x. 2 lerogers users   64 Jan 24 11:30 mydata
drwxr-xr-x. 2 lerogers users   31 Jan 31 10:54 myscripts
[lerogers@pbio381 ~]$ cd fastqcoutput/
[lerogers@pbio381 fastqcoutput]$ ll
total 1384
-rw-r--r--. 1 lerogers users 339080 Jan 29 11:07 WA_PP1_F3_ATGTCA_L003_R1_001_fastqc.html
-rw-r--r--. 1 lerogers users 438237 Jan 29 11:07 WA_PP1_F3_ATGTCA_L003_R1_001_fastqc.zip
-rw-r--r--. 1 lerogers users 285373 Jan 31 11:10 WA_PP1_F3_R1_clean_paired.fa_fastqc.html
-rw-r--r--. 1 lerogers users 349105 Jan 31 11:10 WA_PP1_F3_R1_clean_paired.fa_fastqc.zip


And copied it to our computer in a second terminal window:

Second terminal window: 
Last login: Wed Jan 31 10:57:21 on ttys001
ip0af527d2:~ Lucy$ cd /Users/Lucy/Desktop/Ecological\ Genomics\ Terminal\ Output\ files 
ip0af527d2:Ecological Genomics Terminal Output files Lucy$ scp lerogers@pbio381.uvm.edu:~/fastqcoutput/*html .
lerogers@pbio381.uvm.edu's password: 
WA_PP1_F3_ATGTCA_L003_R1_001_fastqc.html      100%  331KB 331.1KB/s   00:00    
WA_PP1_F3_R1_clean_paired.fa_fastqc.html      100%  279KB 278.7KB/s   00:00    
ip0af527d2:Ecological Genomics Terminal Output files Lucy$ 


Back to first terminal window, we next mapped our reads to the reference transcriptome. It had already been previously downloaded to the directory from a website using the following commands: 
cd /data/project_data/beetles/reference/
wget https://i5k.nal.usda.gov/data/Arthropoda/onttau-%28Onthophagus_taurus%29/Current%20Genome%20Assembly/2.Official%20or%20Primary%20Gene%20Set/BCM_version_0.5.3/consensus_gene_set/OTAU.fna

Then we talked about how we will work within the program bwa to map the directory. The professors had previously indexed the reference transcriptome with the command $ bwa index /data/project_data/beetles/reference/OTAU.fna

Finally, we created a sam directory in our mydata directory. Then we opened up a screen and aligned our sequence that way, such that when we pressed screen -r, the program continued to run but our computer was detached from it. 

[lerogers@pbio381 fastqcoutput]$ cd ~
[lerogers@pbio381 ~]$ cd ~/mydata/
[lerogers@pbio381 mydata]$ mkdir sam
[lerogers@pbio381 mydata]$ ll
total 8
-rw-r--r--. 1 lerogers users  720 Jan 24 11:30 cols_data_Fonly.txt
-rwxr-xr-x. 1 lerogers users 1474 Jan 24 11:25 cols_data.txt
drwxr-xr-x. 2 lerogers users   10 Jan 31 11:28 sam
[lerogers@pbio381 mydata]$ cd sam
[lerogers@pbio381 sam]$ screen
[screen is terminating]
[lerogers@pbio381 sam]$ screen
[detached from 53787.pts-6.pbio381]
[lerogers@pbio381 sam]$ screen -r
[detached from 53787.pts-6.pbio381]
[lerogers@pbio381 sam]$ 



------
<div id='id-section5'/>
### Page 5: 2018-02-05

I am copying the review of the vim and screen tricks from the tutorial because I found them useful:

Review vim and screen tricks
vim is a text editor that can be used from the command line. It can be quite powerful for editing once you know a few tricks. We started with the basics: To open the file trim_myfile.sh

vim trim_myfile.sh 
To edit or “insert” text, type i

To exit the edit mode, hit esc

To save your changes, you “write” the file with :w . Note that you start all commands in vim with a colon. To quit the program, :q .

screen is another fantastically useful program that allows programs or scripts to run in the background if you “detach” from the screen. To open a screen window, type screen. Now you have your command prompt and can start a script or enter a command. To “detach”, type ctrl plus a plus d . To reattach, type  screen -r and the session ID you want to enter.



Then we explored our sam output, then tried to run the python script countexpression_PE and put the output as .txt in the myscripts folder, but it didn't work and I'm not exactly sure why.

Finally, we installed DESeq into RStudio but mine failed because I have to update my computer, R, and RStudio.

------
<div id='id-section6'/>
### Page 6: 2018-02-07

Bash script is a way to keep a record of what you do...I really should review bash, I'm still a bit unclear about it.


[lerogers@pbio381 ~]$ cd ~/myscripts/
[lerogers@pbio381 myscripts]$ ll
total 12
-rwxr--r--. 1 lerogers users 6487 Feb  5 11:18 countxpression_PE.py
-rwxr--r--. 1 lerogers users  764 Jan 31 10:49 trim_F3.sh
[lerogers@pbio381 myscripts]$ vim test_bash
[lerogers@pbio381 myscripts]$ ll
total 16
-rwxr--r--. 1 lerogers users 6487 Feb  5 11:18 countxpression_PE.py
-rw-r--r--. 1 lerogers users   33 Feb  7 10:30 test_bash
-rwxr--r--. 1 lerogers users  764 Jan 31 10:49 trim_F3.sh
[lerogers@pbio381 myscripts]$ chmod u+x test_bash 
[lerogers@pbio381 myscripts]$ ll
total 16
-rwxr--r--. 1 lerogers users 6487 Feb  5 11:18 countxpression_PE.py
-rwxr--r--. 1 lerogers users   33 Feb  7 10:30 test_bash
-rwxr--r--. 1 lerogers users  764 Jan 31 10:49 trim_F3.sh
[lerogers@pbio381 myscripts]$ ./test_bash 
hello world
[lerogers@pbio381 myscripts]$ 


Next, we got a new clean dataset that Melissa had prepared and we switched over to R, where we wrote the code to use DESeq to test for differences in gene expression. 

------
<div id='id-section7'/>
### Page 7: 2018-02-12

We worked on writing our R script for testing for differential gene expression

------
<div id='id-section8'/>
### Page 8: 2018-02-14
More R script work. Here is the script, pasted below:

# set working directory
setwd("~/Desktop/Ecological Genomics Terminal Output files")

library("DESeq2")
library("ggplot2")

countsTable <- read.delim('allcountsdataRN_noIT.txt',header=TRUE, stringsAsFactors = TRUE, row.names = 1)
countData <- as.matrix(countsTable)
head(countData)

conds <- read.delim("cols_data_noIT.txt", header=TRUE, stringsAsFactors = TRUE, row.names = 1)
head(conds)
colData <- as.data.frame(conds)

####################################### we've now set everything up

dds <- DESeqDataSetFromMatrix(countData = countData, colData = colData, design = ~ devstage + sex + population)
# "population effect" model controlling for differences in devstage and sex. Population is listed last, so that's what it's ultimately testing for.

dim(dds)
#make note of output dimenstions: 17483, 48

dds <- dds[rowSums(counts(dds)) > 1,]

dim(dds)
#new dimensions: 16851, 48. Only 632 genes lost.

dds <- DESeq(dds, modelMatrixType = "standard")

resultsNames(dds)
#[1] "Intercept"           "devstage_L3L_vs_AD4" "devstage_PD1_vs_AD4"
#[4] "devstage_PP1_vs_AD4" "sex_M_vs_F"          "population_WA_vs_NC"

res <- results(dds)
str(res)

res <- res[order(res$padj),]
head(res)
#save output table here. use shift+command+C to put many lines as text

# log2 fold change (MLE): population WA vs NC
# Wald test p-value: population WA vs NC
# DataFrame with 6 rows and 6 columns
# baseMean log2FoldChange     lfcSE      stat       pvalue         padj
# <numeric>      <numeric> <numeric> <numeric>    <numeric>    <numeric>
#   OTAU017482-RA 126.291608     -5.4166340 0.6732174 -8.045892 8.561964e-16 1.268369e-11
# OTAU012716-RA 188.877675      4.2644034 0.5427535  7.856980 3.935069e-15 2.914706e-11
# OTAU008667-RA 231.871115     -0.8736955 0.1267134 -6.895051 5.384538e-12 1.994164e-08
# OTAU012562-RA 251.774364     -0.8774079 0.1270957 -6.903520 5.072966e-12 1.994164e-08
# OTAU013988-RA   4.416955      4.4229857 0.6836393  6.469765 9.815559e-11 2.908154e-07
# OTAU011160-RA  10.241516     -2.5149390 0.4125305 -6.096371 1.085033e-09 2.678947e-06

summary(res)

# out of 16851 with nonzero total read count
# adjusted p-value < 0.1
# LFC > 0 (up)     : 333, 2% 
# LFC < 0 (down)   : 282, 1.7% 
# outliers [1]     : 85, 0.5% 
# low counts [2]   : 1952, 12% 
# (mean count < 1)
# [1] see 'cooksCutoff' argument of ?results
# [2] see 'independentFiltering' argument of ?results

#FOR NEXT TIME: what model would you want to do, and how would you set it up???

res_pop <- results(dds, name="population_WA_vs_NC", alpha=0.05)

res_pop <- res_pop[order(res_pop$padj),]
summary(res_pop)

#################### Data Visualization
plotMA(res_pop, main="DESeq2", ylim=c(-2,2))
abline(h=c(-1,1), col="blue", lwd=2)

# sex effect?

res_sex <- results(dds, name="sex_M_vs_F", alpha=0.05)
plotMA(res_sex, main="DESeq2", ylim=c(-4,4))
abline(h=c(-1,1), col="blue", lwd=2)

###### PCA
rld <- rlog(dds, blind=FALSE)
vsd <- vst(dds, blind=FALSE)

data <- plotPCA(vsd, intgroup=c("population", "devstage", "sex"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))

data$devstage <- factor(data$devstage, levels=c("L3L", "PP1", "PD1", "AD4"), labels = c("L3L", "PP1", "PD1", "AD4"))

pdf(file="PCA_sex_by_stage.pdf", height=5.5, width=5.5)
ggplot(data, aes(PC1, PC2, color=sex, shape=devstage))  + 
  geom_point(size=4, alpha=0.85) +
  xlab(paste0("PC1: ",percentVar[1], "% variance")) +
  ylab(paste0("PC2: ",percentVar[2], "% variance")) +
  theme_minimal()  
dev.off()                        

ggplot(data, aes(PC1, PC2, color=population, shape=devstage))  + 
  geom_point(size=4, alpha=0.85) +
  xlab(paste0("PC1: ",percentVar[1], "% variance")) +
  ylab(paste0("PC2: ",percentVar[2], "% variance")) +
  theme_minimal()  






## the next several lines will make a cluster heatmap of all the samples vs all the samples

sampleDists <- dist(t(assay(rld)))

library("RColorBrewer")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(rld$devstage, rld$population, rld$sex, sep="-")
colnames(sampleDistMatrix) <- NULL

colors <- colorRampPalette(rev(brewer.pal(9,"Greens")))(255)

library("pheatmap")

pheatmap(sampleDistMatrix,clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col=colors)

#In general, it's good practice to load all the libraries at the top at first, so you can run chunks of code separately

#################Let's look at individual genes!!

d <- plotCounts(dds, gene = "OTAU017482-RA", intgroup = (c("population","sex","devstage")), returnData=TRUE)
d

p <-ggplot(d, aes(x=devstage, y=count, shape=sex, colour = population)) +
  theme_minimal() + theme(text = element_text(size=20), panel.grid.major=element_line(colour="grey"))
p <- p + geom_point(position = position_jitter(w=0.3,h=0), size=3) +
  scale_x_discrete(limits=c("L3L","PP1","PD1","AD4"))
p

################ Save outputs

write.csv(res_pop, file="DGE_NCvsWA_pop_classDay6.csv", row.names = T, quote=F)


### change pvalue to -log(pvalue) and export as .csv with rownames

neglogpval <- as.matrix(-log(res_pop$pvalue))
head(neglogpval)

res_pop_negpval <- cbind(row.names(res_pop),neglogpval)
head(res_pop_negpval)

colnames(res_pop_negpval)=c("gene","neglogpval")

write.csv(res_pop_negpval, file="DGE_NCvsWA_pop_neglogpval_classDay6.csv", row.names=F,quote=F)

------
<div id='id-section9'/>
### Page 9:2018-02-21

Notes on my question about bash script: allows you to write multiple commands at once and then send them to the server all at once.  Allows you to document your command all at once, also allows you to go in screen mode and then leave your computer for a week and have it be running things while you disconnect from you.  When you use command line you are also using bash, but bash script is when you open up a new screen and write a script. UNIX can also understand python scripts.

We found that the DESeq version is different and is segregating the class between version 1.18.1 and 1.14.1.  This is a lesson about the importance of reporting your version when you are writing a paper.

Using R, we made a heat map comparing WAF, WAM, NCF, NCM.   Here is the new script we wrote: 

# set working directory
setwd("~/Desktop/Ecological Genomics Terminal Output files")

library("DESeq2")
library("ggplot2")

countsTable <- read.delim('allcountsdataRN_noIT.txt',header=TRUE, stringsAsFactors = TRUE, row.names = 1)
countData <- as.matrix(countsTable)
head(countData)

conds <- read.delim("cols_data_noIT.txt", header=TRUE, stringsAsFactors = TRUE, row.names = 1)
head(conds)
colData <- as.data.frame(conds)

############################# Day 7 Transcriptomics


colData$group <- factor(paste0(colData$population, "-", colData$devstage, "-", colData$sex))
head(colData)

dds <- DESeqDataSetFromMatrix(countData = countData, colData = colData, design = ~ group)
dds <- dds[rowSums(counts(dds)) > 1, ]
dim(dds)

dds <- DESeq(dds, modelMatrixType = "standard", parallel =T)
#dds <- DESeq(dds, modelMatrixType = "expanded", parallel=T)

resultsNames(dds)
# [1] "Intercept"                  "group_NC.AD4.M_vs_NC.AD4.F" "group_NC.L3L.F_vs_NC.AD4.F" "group_NC.L3L.M_vs_NC.AD4.F"
# [5] "group_NC.PD1.F_vs_NC.AD4.F" "group_NC.PD1.M_vs_NC.AD4.F" "group_NC.PP1.F_vs_NC.AD4.F" "group_NC.PP1.M_vs_NC.AD4.F"
# [9] "group_WA.AD4.F_vs_NC.AD4.F" "group_WA.AD4.M_vs_NC.AD4.F" "group_WA.L3L.F_vs_NC.AD4.F" "group_WA.L3L.M_vs_NC.AD4.F"
# [13] "group_WA.PD1.F_vs_NC.AD4.F" "group_WA.PD1.M_vs_NC.AD4.F" "group_WA.PP1.F_vs_NC.AD4.F" "group_WA.PP1.M_vs_NC.AD4.F"

res_pop_PP1_F <- results(dds, contrast = list(c("group_NC.PP1.F_vs_NC.AD4.F"),c("group_WA.PP1.F_vs_NC.AD4.F")), listValues = c(1/2,-1/2), alpha = 0.05)

res_pop_PP1_F <-res_pop_PP1_F[order(res_pop_PP1_F$padj),]
head(res_pop_PP1_F)
# log2 fold change (MLE): 0.5 group_NC.PP1.F_vs_NC.AD4.F vs 0.5 group_WA.PP1.F_vs_NC.AD4.F 
# Wald test p-value: 0.5 group_NC.PP1.F_vs_NC.AD4.F vs 0.5 group_WA.PP1.F_vs_NC.AD4.F 
# DataFrame with 6 rows and 6 columns
# baseMean log2FoldChange     lfcSE       stat       pvalue         padj
# <numeric>      <numeric> <numeric>  <numeric>    <numeric>    <numeric>
#   OTAU012622-RA   31.61690     -24.015012  1.336019 -17.975053 3.055737e-72 4.245641e-68
# OTAU009289-RA   19.87268      16.377873  1.973838   8.297477 1.063460e-16 7.387858e-13
# OTAU007155-RA 2989.82336     -10.440891  1.286915  -8.113117 4.933741e-16 2.284980e-12
# OTAU012427-RA   51.36362       9.162012  1.149377   7.971287 1.570307e-15 5.454462e-12
# OTAU010169-RA   12.03314       9.613149  1.243946   7.727950 1.092923e-14 3.037015e-11
# OTAU011364-RA 2094.24323       8.803863  1.208617   7.284246 3.234734e-13 7.490565e-10

summary(res_pop_PP1_F)
# out of 16851 with nonzero total read count
# adjusted p-value < 0.05
# LFC > 0 (up)     : 46, 0.27% 
# LFC < 0 (down)   : 31, 0.18% 
# outliers [1]     : 17, 0.1% 
# low counts [2]   : 2940, 17% 
# (mean count < 3)

############# Pull out significant genes to be able to make a heatmap

sig_pop_PP1_F <- res_pop_PP1_F[which(res_pop_PP1_F$padj <0.05), ]  # great way to subset data
dim(sig_pop_PP1_F) # check

sig_pop_PP1_F_df <- as.data.frame(sig_pop_PP1_F)
sig_pop_PP1_F_df$Row.names <- rownames(sig_pop_PP1_F_df)
dim(sig_pop_PP1_F_df) #column dimensions increase by one because we made the rownames a new column

genesOfInterest_pop_PP1_F <- c(sig_pop_PP1_F_df$Row.names)
length(genesOfInterest_pop_PP1_F)

vsd <- vst(dds, blind =FALSE)

####

dds$combined = factor(paste0(dds$population, "-", dds$devstage, "-", dds$sex))
dds$combined <- factor(dds$combined, levels=c("WA-L3L-F","WA-L3L-M","WA-PP1-F","WA-PP1-M","WA-PD1-F","WA-PD1-M","WA-AD4-F","WA-AD4-M","NC-L3L-F","NC-L3L-M","NC-PP1-F","NC-PP1-M","NC-PD1-F","NC-PD1-M","NC-AD4-F","NC-AD4-M"), labels=c("WA-L3L-F","WA-L3L-M","WA-PP1-F","WA-PP1-M","WA-PD1-F","WA-PD1-M","WA-AD4-F","WA-AD4-M","NC-L3L-F","NC-L3L-M","NC-PP1-F","NC-PP1-M","NC-PD1-F","NC-PD1-M","NC-AD4-F","NC-AD4-M"))

baseMeanPerGrp <- sapply( levels(dds$combined), function(lvl) rowMeans( counts(dds,normalized=TRUE)[,dds$combined == lvl] ) )


head(baseMeanPerGrp)
dim(baseMeanPerGrp)

# pulls out normalized counts (avg of 3 reps) for all of our significant genes
m <- baseMeanPerGrp[genesOfInterest_pop_PP1_F, c("WA-PP1-F", "WA-PP1-M", "NC-PP1-F", "NC-PP1-M")]

head(m)
dim(m)

mat_scaled = t(apply(m, 1, scale)) #gives them all mean of 0 and SD of 1
head(mat_scaled)

library(pheatmap)

pheatmap(mat_scaled, labels_col = c("WA-PP1-F", "WA-PP1-M", "NC-PP1-F", "NC-PP1-M"), cluster_cols = T, cluster_rows = T)

####################

# Export counts to use as the input for WGCNA

norm.counts <- counts(dds, normalized=TRUE)
dim(norm.counts)

write.csv(norm.counts, file = "beetle_norm_counts.csv", row.names = T, quote=F)





Then we started a new R script to install the package WGCNA and look at module-trait relationships.

Melissa emailed us a new file which recodes the data table to be just numerical data, not categorical like 'male' or 'female', and we put that into our library.

The script is below:


source("http://bioconductor.org/biocLite.R") 
biocLite(c("AnnotationDbi", "impute", "GO.db", "preprocessCore")) 
install.packages("WGCNA")

library("WGCNA")

setwd("~/Desktop/Ecological Genomics Terminal Output files")

beetData = read.csv("beetle_norm_counts.csv")
dim(beetData)
head(beetData)
names(beetData)


bdatExpr0 = as.data.frame(t(beetData[, -c(1:1)])) #t means transpose; -c(1:1) means remove column 1
dim(bdatExpr0)
names(bdatExpr0)= beetData$X

rownames(bdatExpr0) = names(beetData)[-c(1:1)]

bgsg = goodSamplesGenes(bdatExpr0, verbose = 3) #bgsg = beetles good samples genes

########### Make a cluster tree to look for outlier samples

sampleTree = hclust(dist(bdatExpr0), method= "average")
## plot
sizeGrWindow(12,9)
par(cex = 0.6)
par(mar=c(0,4,2,0))

plot(sampleTree, main = 'Sample clustering to detect outliers', sub="", xlab="", cex.lab = 1.5, cex.axis = 1.5, cex.main = 2)


####### Read in trait data

btraitData = read.table("cols_data_noIT_num.txt", header=T)
dim(btraitData)
head(btraitData)

###### Network construction and module detection

powers = c(c(1:10), seq(from = 12, to =20, by = 2))
#Call the network topology function

sft = pickSoftThreshold(bdatExpr0, powerVector = powers, verbose=5)

#Plot results
sizeGrWindow(9,5)
par(mfrow = c(1,2))
cex1 = 0.9

# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2], xlab="Soft Threshold (power)", ylab="Scale Free Topology Model Git, signed R^2", type = "n", main=paste("Scale independence"))

text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2], labels=powers, cex=1, col="red")

abline(h=0.9, col="red")

# mean connectivity as a function of the soft-thresholding power

plot(sft$fitIndices[,1], sft$fitIndices[,5], xlab="Soft Threshold (power)", ylab = "Mean Connectivity", type = "n", main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1, col="red")


#######

net = blockwiseModules(bdatExpr0, power=6, TOMType = "unsigned", minModuleSize = 30, reassignThreshold = 0, mergeCutHeight = 0.25, numericLabels = T, pamRespectsDendro = F, saveTOMs = T, saveTOMFileBase = "beetleTOM", verbose = 3)
#TOM = topology modules
sizeGrWindow(12,9)
mergedColors = labels2colors(net$colors)

## Plot the dendrogram with the module colors underneath
plotDendroAndColors(net$dendrograms[[1]],mergedColors[net$blockGenes[[1]]], "Module colors", dendroLables = F, hang = 0.03, addGuide = T, guideHang = 0.05)


moduleLabels = net$colors
moduleColors = labels2colors(net$colors)
MEs = net$MEs #Module Eigengenes
geneTree = net$dendrograms[[1]]

### Define numbers of genes and samples

nGenes = ncol(bdatExpr0)
nSamples= nrow(bdatExpr0)

MEs0 = moduleEigengenes(bdatExpr0, moduleColors)$eigengenes
MEs = orderMEs(MEs0)
moduleTraitCor = cor(MEs, btraitData, use = "p")

moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples)

sizeGrWindow(10,6)
textMatrix = paste(signif(moduleTraitCor, 2), "\n(", signif(moduleTraitPvalue,1), ")", sep = "")
dim(textMatrix) = dim(moduleTraitCor)
par(mar = c(6, 8.5, 3, 3))
labeledHeatmap(Matrix = moduleTraitCor, xLabels = names(btraitData), yLabels = names(MEs), ySymbols = names(MEs), colorLabels = F, colors = greenWhiteRed(50), textMatrix = textMatrix, setStdMargins = F, cex.text = 0.5, zlim = c(-1,1), main = paste("Module-trait relationships"))

------
<div id='id-section10'/>
### Page 10: 2-26-2018: Intro to SNP and Genotype Calling

Right now our sam file is human readable, we are going to write a bash script so we can let it run without us babysitting it and put it into bam. My sam file is called WA_PP1_F3_bwamem.sam and is in my sam folder, inside my mydata folder.

Once we open up vim, we type i to start typing and then #! to tell it how to interpret what we are saying.  Here is the script: 



#!/bin/bash

#Convert sam file to binary bam format

samtools view -b -@ 4 WA_PP1_F3_bwamem.sam -o WA_PP1_F3_bwamem.bam

# Check stats on our alignment

samtools flagstat WA_PP1_F3_bwamem.bam

samtools depth WA_PP1_F3_bwamem.bam

# Fix mate pairs between paired end reads that have been orphaned

samtools fixmate WA_PP1_F3_bwamem.bam WA_PP1_F3_bwamem.fixmate.bam

# Sort the alignments
samtools sort -@ 4 WA_PP1_F3_bwamem.fixmate.bam WA_PP1_F3_bwamem.fixmate.sorted.bam

# Remove PCR duplicates
#We are not actually doing this for time's sake, but here's what it would look like:
# samtools rmdup WA_PP1_F3_bwamem.fixmate.sorted.bam WA_PP1_F3_bwamem.fixmate.sorted.rmdup.bam

# Index our final alignment for fast processing

samtools index WA_PP1_F3_bwamem.fixmate.sorted.bam

# Pause here; look at the results before proceeding to SNP calling



then we did esc to leave insert mode, and :w with the name to name/save, and :q to quit. The script was then saved to my sam folder, which is inside my mydata folder.  We made it executable by adding chmod o+x CallSNPs.sh 

Then, we typed 'screen' to enter a new screen and run the bash script by typing 
bash CallSNPs.sh 
We got no response and no command prompt, so we know it's running. So we detach with ctr-a-d


Here's the whole script from today:

Last login: Wed Feb 21 01:05:17 on ttys000
ip0af52072:~ Lucy$ ssh lerogers@pbio381.uvm.edu
lerogers@pbio381.uvm.edu's password: 
Last login: Wed Feb 21 01:05:32 2018 from ip040077.uvm.edu
[lerogers@pbio381 ~]$ ll
total 8
drwxr-xr-x. 2 lerogers users 4096 Jan 31 10:55 cleanreads
drwxr-xr-x. 2 lerogers users 4096 Jan 31 11:10 fastqcoutput
drwxr-xr-x. 3 lerogers users   78 Jan 31 11:28 mydata
drwxr-xr-x. 2 lerogers users   82 Feb  7 10:30 myscripts
[lerogers@pbio381 ~]$ cd mydata
[lerogers@pbio381 mydata]$ ll
total 12
-rw-r--r--. 1 lerogers users  720 Jan 24 11:30 cols_data_Fonly.txt
-rwxr-xr-x. 1 lerogers users 1474 Jan 24 11:25 cols_data.txt
drwxr-xr-x. 2 lerogers users 4096 Feb  5 11:16 sam
[lerogers@pbio381 mydata]$ vim
[lerogers@pbio381 mydata]$ cd sam
[lerogers@pbio381 sam]$ ll
total 6221228
-rw-r--r--. 1 lerogers users        195 Feb  5 11:17 countstatsummary.txt
-rw-r--r--. 1 lerogers users      27155 Feb  5 10:51 tail.sam
-rw-r--r--. 1 lerogers users     349720 Feb  5 11:17 WA_PP1_F3_bwamem_counts.txt
-rw-r--r--. 1 lerogers users 6370145913 Jan 31 12:02 WA_PP1_F3_bwamem.sam
[lerogers@pbio381 sam]$ head WA_PP1_F3_bwamem.sam
@SQ	SN:OTAU000001-RA	LN:3192
@SQ	SN:OTAU000002-RA	LN:4557
@SQ	SN:OTAU000003-RA	LN:1051
@SQ	SN:OTAU000004-RA	LN:735
@SQ	SN:OTAU000005-RA	LN:456
@SQ	SN:OTAU000006-RA	LN:996
@SQ	SN:OTAU000007-RA	LN:261
@SQ	SN:OTAU000008-RA	LN:15102
@SQ	SN:OTAU000009-RA	LN:3352
@SQ	SN:OTAU000010-RA	LN:6187
[lerogers@pbio381 sam]$ vim




[lerogers@pbio381 sam]$ ll
total 6221232
-rw-r--r--. 1 lerogers users        848 Feb 26 11:03 CallSNPs.sh
-rw-r--r--. 1 lerogers users        195 Feb  5 11:17 countstatsummary.txt
-rw-r--r--. 1 lerogers users      27155 Feb  5 10:51 tail.sam
-rw-r--r--. 1 lerogers users     349720 Feb  5 11:17 WA_PP1_F3_bwamem_counts.txt
-rw-r--r--. 1 lerogers users 6370145913 Jan 31 12:02 WA_PP1_F3_bwamem.sam
[lerogers@pbio381 sam]$ ch mod o+x CallSNPs.sh 
-bash: ch: command not found
[lerogers@pbio381 sam]$ chmod o+x CallSNPs.sh 
[lerogers@pbio381 sam]$ ll
total 6221232
-rw-r--r-x. 1 lerogers users        848 Feb 26 11:03 CallSNPs.sh
-rw-r--r--. 1 lerogers users        195 Feb  5 11:17 countstatsummary.txt
-rw-r--r--. 1 lerogers users      27155 Feb  5 10:51 tail.sam
-rw-r--r--. 1 lerogers users     349720 Feb  5 11:17 WA_PP1_F3_bwamem_counts.txt
-rw-r--r--. 1 lerogers users 6370145913 Jan 31 12:02 WA_PP1_F3_bwamem.sam
[lerogers@pbio381 sam]$ screen
[detached from 128482.pts-5.pbio381]
[lerogers@pbio381 sam]$ top

top - 11:10:43 up 34 days, 19:30,  7 users,  load average: 6.84, 5.28, 2.28
Tasks: 330 total,   7 running, 323 sleeping,   0 stopped,   0 zombie
%Cpu(s): 20.4 us,  0.8 sy,  0.0 ni, 78.3 id,  0.4 wa,  0.0 hi,  0.0 si,  0.0 st
KiB Mem : 32764504 total,   221872 free,   474100 used, 32068532 buff/cache
KiB Swap:  2097148 total,  2095348 free,     1800 used. 31607496 avail Mem 

   PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                
128786 msouthga  20   0   25100   2628    876 R 100.0  0.0   1:22.80 samtools                               
128763 srkeller  20   0   25108   2640    888 R 100.0  0.0   2:00.40 samtools                               
128782 pcronin1  20   0   24968   2628    876 R 100.0  0.0   1:23.77 samtools                               
128784 lerogers  20   0   25100   2628    876 R 100.0  0.0   1:23.10 samtools                               
128780 eathibau  20   0   25100   2628    876 R  99.3  0.0   1:29.42 samtools                               
 34800 root      20   0       0      0      0 S   6.0  0.0   0:16.56 kworker/u256:2                         
 75822 root      20   0       0      0      0 S   1.3  0.0   0:19.01 kworker/1:2                            
   140 root      20   0       0      0      0 S   0.7  0.0  21:53.00 kswapd0                                
 24396 root      20   0       0      0      0 S   0.3  0.0   0:17.24 kworker/0:1                            
128788 srkeller  20   0  159948   2440   1560 S   0.3  0.0   0:00.17 top                                    
128792 lerogers  20   0  159956   2484   1564 R   0.3  0.0   0:00.10 top                                    
     1 root      20   0  193580   5860   3296 S   0.0  0.0   3:55.49 systemd                                
     2 root      20   0       0      0      0 S   0.0  0.0   0:06.62 kthreadd                               
     3 root      20   0       0      0      0 S   0.0  0.0   0:03.37 ksoftirqd/0                            
     5 root       0 -20       0      0      0 S   0.0  0.0   0:00.00 kworker/0:0H                           
     7 root      rt   0       0      0      0 S   0.0  0.0   0:00.94 migration/0                            
     8 root      20   0       0      0      0 S   0.0  0.0   0:00.00 rcu_bh                                 
     9 root      20   0       0      0      0 R   0.0  0.0  10:55.70 rcu_sched                              
    10 root      rt   0       0      0      0 S   0.0  0.0   0:05.35 watchdog/0                             
    11 root      rt   0       0      0      0 S   0.0  0.0   0:08.71 watchdog/1                             
    12 root      rt   0       0      0      0 S   0.0  0.0   0:01.63 migration/1                            
    13 root      20   0       0      0      0 S   0.0  0.0   0:00.61 ksoftirqd/1                            
    15 root       0 -20       0      0      0 S   0.0  0.0   0:00.00 kworker/1:0H                           
    16 root      rt   0       0      0      0 S   0.0  0.0   0:07.25 watchdog/2                             
    17 root      rt   0       0      0      0 S   0.0  0.0   0:02.14 migration/2                            
    18 root      20   0       0      0      0 S   0.0  0.0   0:00.53 ksoftirqd/2                            
    20 root       0 -20       0      0      0 S   0.0  0.0   0:00.00 kworker/2:0H                           
    21 root      rt   0       0      0      0 S   0.0  0.0   0:07.98 watchdog/3                             
    22 root      rt   0       0      0      0 S   0.0  0.0   0:01.74 migration/3                            
    23 root      20   0       0      0      0 S   0.0  0.0   0:00.49 ksoftirqd/3                            
    25 root       0 -20       0      0      0 S   0.0  0.0   0:00.00 kworker/3:0H                           
    26 root      rt   0       0      0      0 S   0.0  0.0   0:06.61 watchdog/4                             
[lerogers@pbio381 sam]$ ll
total 9245240
-rw-r--r-x. 1 lerogers users        848 Feb 26 11:03 CallSNPs.sh
-rw-r--r--. 1 lerogers users        195 Feb  5 11:17 countstatsummary.txt
-rw-r--r--. 1 lerogers users      27155 Feb  5 10:51 tail.sam
-rw-r--r--. 1 lerogers users 2022906003 Feb 26 11:08 WA_PP1_F3_bwamem.bam
-rw-r--r--. 1 lerogers users     349720 Feb  5 11:17 WA_PP1_F3_bwamem_counts.txt
-rw-r--r--. 1 lerogers users  584278169 Feb 26 11:10 WA_PP1_F3_bwamem.fixmate.bam
-rw-r--r--. 1 lerogers users 6370145913 Jan 31 12:02 WA_PP1_F3_bwamem.sam
[lerogers@pbio381 sam]$ vi CallSNPs.sh 
[lerogers@pbio381 sam]$ screen -r
[detached from 128482.pts-5.pbio381]
[lerogers@pbio381 sam]$ samtools sort -@ 4 WA_PP1_F3_bwamem.fixmate.bam  -o bash CallSNPs.sh 
[bam_sort] Use -T PREFIX / -o FILE to specify temporary and final output files
Usage: samtools sort [options...] [in.bam]
Options:
  -l INT     Set compression level, from 0 (uncompressed) to 9 (best)
  -m INT     Set maximum memory per thread; suffix K/M/G recognized [768M]
  -n         Sort by read name
  -o FILE    Write final output to FILE rather than standard output
  -T PREFIX  Write temporary files to PREFIX.nnnn.bam
      --input-fmt-option OPT[=VAL]
               Specify a single input file format option in the form
               of OPTION or OPTION=VALUE
  -O, --output-fmt FORMAT[,OPT[=VAL]]...
               Specify output format (SAM, BAM, CRAM)
      --output-fmt-option OPT[=VAL]
               Specify a single output file format option in the form
               of OPTION or OPTION=VALUE
      --reference FILE
               Reference sequence FASTA FILE [null]
  -@, --threads INT
               Number of additional threads to use [0]
[lerogers@pbio381 sam]$ 
[lerogers@pbio381 sam]$ samtools sort -@ 4 WA_PP1_F3_bwamem.fixmate.bam  -o WA_PP1_F1.fixmate.sorted.vam
[bam_sort_core] merging from 12 files...
[lerogers@pbio381 sam]$ ll
total 11681652
-rw-r--r-x. 1 lerogers users        848 Feb 26 11:03 CallSNPs.sh
-rw-r--r--. 1 lerogers users        195 Feb  5 11:17 countstatsummary.txt
-rw-r--r--. 1 lerogers users      27155 Feb  5 10:51 tail.sam
-rw-r--r--. 1 lerogers users 1497739702 Feb 26 11:19 WA_PP1_F1.fixmate.sorted.vam
-rw-r--r--. 1 lerogers users 2022906003 Feb 26 11:08 WA_PP1_F3_bwamem.bam
-rw-r--r--. 1 lerogers users     349720 Feb  5 11:17 WA_PP1_F3_bwamem_counts.txt
-rw-r--r--. 1 lerogers users 2070814941 Feb 26 11:14 WA_PP1_F3_bwamem.fixmate.bam
-rw-r--r--. 1 lerogers users 6370145913 Jan 31 12:02 WA_PP1_F3_bwamem.sam
[lerogers@pbio381 sam]$ samtools index WA_PP1_F1.fixmate.sorted.vam
[lerogers@pbio381 sam]$ ll
total 11682908
-rw-r--r-x. 1 lerogers users        848 Feb 26 11:03 CallSNPs.sh
-rw-r--r--. 1 lerogers users        195 Feb  5 11:17 countstatsummary.txt
-rw-r--r--. 1 lerogers users      27155 Feb  5 10:51 tail.sam
-rw-r--r--. 1 lerogers users 1497739702 Feb 26 11:19 WA_PP1_F1.fixmate.sorted.vam
-rw-r--r--. 1 lerogers users    1282176 Feb 26 11:21 WA_PP1_F1.fixmate.sorted.vam.bai
-rw-r--r--. 1 lerogers users 2022906003 Feb 26 11:08 WA_PP1_F3_bwamem.bam
-rw-r--r--. 1 lerogers users     349720 Feb  5 11:17 WA_PP1_F3_bwamem_counts.txt
-rw-r--r--. 1 lerogers users 2070814941 Feb 26 11:14 WA_PP1_F3_bwamem.fixmate.bam
-rw-r--r--. 1 lerogers users 6370145913 Jan 31 12:02 WA_PP1_F3_bwamem.sam
[lerogers@pbio381 sam]$ samtools tview WA_PP1_F1.fixmate.sorted.vam
[lerogers@pbio381 sam]$ samtools tview WA_PP1_F1.fixmate.sorted.vam
[lerogers@pbio381 sam]$ samtools tview WA_PP1_F1.fixmate.sorted.vam /data/project_data/beetles/reference/OTAU.fna
[lerogers@pbio381 sam]$ 





------
<div id='id-section11'/>
### Page 11: 2018-2-28: Diversity and Site Frequency Spectrum (SFS)

In vim text editor, use :wq to write your script and then quit; essentially to save the edits.

In vim when looking at output, use :set nowrap to see the data and :1900 to get below the header.

The genotype sites that have no data is because it didn't pass the min DP = 5 or the min GP = 0.9 threshholds.

We are using files that are in the common space, but outputting files into our home space. For future reference, this is the directory where the data live: /data/project_data/beetles/snps


Then we use the program vcftools to work with the data on our site

We use bi-allelic filter to look at only bi-allelic sites

Setting minGQ to 10 didn't get rid of any results, it just set the lower quality ones to missing data, shown by a dot, instead of getting rid of them. So now we look at what % at a site have information that has passed the quality filter

max-missing 0.8 means it only allows SNPs with at least 80% of the data present

maf = minor allele frequency filter

last thing we did was save a new file after we'd filtered out these filters

So now we saved the vcf output file to our home directory in the myresults file.

Last login: Mon Feb 26 10:42:52 on ttys001
ip0af5234b:~ Lucy$ lerogers@pbio381.uvm.edu
-bash: lerogers@pbio381.uvm.edu: command not found
ip0af5234b:~ Lucy$ ssh lerogers@pbio381.uvm.edu
lerogers@pbio381.uvm.edu's password: 
Last login: Mon Feb 26 10:41:46 2018 from ip0af52072.int.uvm.edu
[lerogers@pbio381 ~]$ cd ~/mydata
[lerogers@pbio381 mydata]$ ll
total 12
-rw-r--r--. 1 lerogers users  720 Jan 24 11:30 cols_data_Fonly.txt
-rwxr-xr-x. 1 lerogers users 1474 Jan 24 11:25 cols_data.txt
drwxr-xr-x. 2 lerogers users 4096 Feb 26 11:21 sam
[lerogers@pbio381 mydata]$ cd sam/
[lerogers@pbio381 sam]$ ll
total 11682908
-rw-r--r-x. 1 lerogers users        848 Feb 26 11:03 CallSNPs.sh
-rw-r--r--. 1 lerogers users        195 Feb  5 11:17 countstatsummary.txt
-rw-r--r--. 1 lerogers users      27155 Feb  5 10:51 tail.sam
-rw-r--r--. 1 lerogers users 1497739702 Feb 26 11:19 WA_PP1_F1.fixmate.sorted.vam
-rw-r--r--. 1 lerogers users    1282176 Feb 26 11:21 WA_PP1_F1.fixmate.sorted.vam.bai
-rw-r--r--. 1 lerogers users 2022906003 Feb 26 11:08 WA_PP1_F3_bwamem.bam
-rw-r--r--. 1 lerogers users     349720 Feb  5 11:17 WA_PP1_F3_bwamem_counts.txt
-rw-r--r--. 1 lerogers users 2070814941 Feb 26 11:14 WA_PP1_F3_bwamem.fixmate.bam
-rw-r--r--. 1 lerogers users 6370145913 Jan 31 12:02 WA_PP1_F3_bwamem.sam
[lerogers@pbio381 sam]$ vim CallSNPs.sh 
[lerogers@pbio381 sam]$ samtools flagstat WA_PP1_F1.fixmate.sorted.vam
22793375 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 secondary
79419 + 0 supplementary
0 + 0 duplicates
18462048 + 0 mapped (81.00% : N/A)
22713956 + 0 paired in sequencing
11356978 + 0 read1
11356978 + 0 read2
17527234 + 0 properly paired (77.17% : N/A)
17965134 + 0 with itself and mate mapped
417495 + 0 singletons (1.84% : N/A)
310662 + 0 with mate mapped to a different chr
261021 + 0 with mate mapped to a different chr (mapQ>=5)
[lerogers@pbio381 sam]$ vim CallSNPs.sh 
[lerogers@pbio381 sam]$ vim CallSNPs.sh 
[lerogers@pbio381 sam]$ bash CallSNPs.sh 
Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid
Could not parse tag "" in "AD,"
Failed to open -: unknown file type
[lerogers@pbio381 sam]$ vim CallSNPs.sh 
[lerogers@pbio381 sam]$ bash CallSNPs.sh 
Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid
Could not parse tag "" in "AD,"
Failed to open -: unknown file type
[lerogers@pbio381 sam]$ vim CallSNPs.sh 
[lerogers@pbio381 sam]$ vim CallSNPs.sh 
[lerogers@pbio381 sam]$ ll
total 11682908
-rw-r--r-x. 1 lerogers users       1289 Feb 28 10:46 CallSNPs.sh
-rw-r--r--. 1 lerogers users        195 Feb  5 11:17 countstatsummary.txt
-rw-r--r--. 1 lerogers users      27155 Feb  5 10:51 tail.sam
-rw-r--r--. 1 lerogers users 1497739702 Feb 26 11:19 WA_PP1_F1.fixmate.sorted.vam
-rw-r--r--. 1 lerogers users    1282176 Feb 26 11:21 WA_PP1_F1.fixmate.sorted.vam.bai
-rw-r--r--. 1 lerogers users 2022906003 Feb 26 11:08 WA_PP1_F3_bwamem.bam
-rw-r--r--. 1 lerogers users     349720 Feb  5 11:17 WA_PP1_F3_bwamem_counts.txt
-rw-r--r--. 1 lerogers users 2070814941 Feb 26 11:14 WA_PP1_F3_bwamem.fixmate.bam
-rw-r--r--. 1 lerogers users 6370145913 Jan 31 12:02 WA_PP1_F3_bwamem.sam
-rw-r--r--. 1 lerogers users          0 Feb 28 10:46 WA_PP1_F3.vcf
[lerogers@pbio381 sam]$ mv WA_PP1_F1.fixmate.sorted.vam WA_PP1_F1.fixmate.sorted.bam
[lerogers@pbio381 sam]$ ll
total 11682908
-rw-r--r-x. 1 lerogers users       1289 Feb 28 10:46 CallSNPs.sh
-rw-r--r--. 1 lerogers users        195 Feb  5 11:17 countstatsummary.txt
-rw-r--r--. 1 lerogers users      27155 Feb  5 10:51 tail.sam
-rw-r--r--. 1 lerogers users 1497739702 Feb 26 11:19 WA_PP1_F1.fixmate.sorted.bam
-rw-r--r--. 1 lerogers users    1282176 Feb 26 11:21 WA_PP1_F1.fixmate.sorted.vam.bai
-rw-r--r--. 1 lerogers users 2022906003 Feb 26 11:08 WA_PP1_F3_bwamem.bam
-rw-r--r--. 1 lerogers users     349720 Feb  5 11:17 WA_PP1_F3_bwamem_counts.txt
-rw-r--r--. 1 lerogers users 2070814941 Feb 26 11:14 WA_PP1_F3_bwamem.fixmate.bam
-rw-r--r--. 1 lerogers users 6370145913 Jan 31 12:02 WA_PP1_F3_bwamem.sam
-rw-r--r--. 1 lerogers users          0 Feb 28 10:46 WA_PP1_F3.vcf
[lerogers@pbio381 sam]$ bash CallSNPs.sh
Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid
Could not parse tag "" in "AD,"
Failed to open -: unknown file type
[lerogers@pbio381 sam]$ mv WA_PP1_F1.fixmate.sorted.vam.bai WA_PP1_F1.fixmate.sorted.bam.bai
[lerogers@pbio381 sam]$ bash CallSNPs.sh 
Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid
Could not parse tag "" in "AD,"
Failed to open -: unknown file type
[lerogers@pbio381 sam]$ vim CallSNPs.sh 
[lerogers@pbio381 sam]$ bash CallSNPs.sh 
Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid
[mpileup] 1 samples in 1 input files
[lerogers@pbio381 sam]$ ll
total 11724832
-rw-r--r-x. 1 lerogers users       1288 Feb 28 10:50 CallSNPs.sh
-rw-r--r--. 1 lerogers users        195 Feb  5 11:17 countstatsummary.txt
-rw-r--r--. 1 lerogers users      27155 Feb  5 10:51 tail.sam
-rw-r--r--. 1 lerogers users 1497739702 Feb 26 11:19 WA_PP1_F1.fixmate.sorted.bam
-rw-r--r--. 1 lerogers users    1282176 Feb 26 11:21 WA_PP1_F1.fixmate.sorted.bam.bai
-rw-r--r--. 1 lerogers users 2022906003 Feb 26 11:08 WA_PP1_F3_bwamem.bam
-rw-r--r--. 1 lerogers users     349720 Feb  5 11:17 WA_PP1_F3_bwamem_counts.txt
-rw-r--r--. 1 lerogers users 2070814941 Feb 26 11:14 WA_PP1_F3_bwamem.fixmate.bam
-rw-r--r--. 1 lerogers users 6370145913 Jan 31 12:02 WA_PP1_F3_bwamem.sam
-rw-r--r--. 1 lerogers users   42928107 Feb 28 11:03 WA_PP1_F3.vcf
[lerogers@pbio381 sam]$ vim WA_PP1_F3.vcf
[lerogers@pbio381 sam]$ vim WA_PP1_F3.vcf
[lerogers@pbio381 sam]$ vim CallSNPs.sh 
[lerogers@pbio381 sam]$ cd /data/project_data/beetles/snps
[lerogers@pbio381 snps]$ ll
total 6369252
-rw-r--r--. 1 srkeller users 4980896352 Feb 27 11:06 OTAU_2018_reads2snps.fas
-rw-r--r--. 1 srkeller users  215746604 Feb 27 13:24 OTAU_2018_reads2snps.vcf
-rw-r--r--. 1 srkeller users 1325465210 Feb 27 16:51 OTAU_2018_samtools.vcf
[lerogers@pbio381 snps]$ tail OTAU_2018_samtools.vcf 
OTAU017482-RA	806	.	C	A	19.8295	.	DP=219;VDB=0.0665622;SGB=-0.118132;RPB=0.138686;MQB=0.00336062;MQSB=0.911687;BQB=0.0212898;MQ0F=0;ICB=0.00120955;HOB=0.00059453;AC=1;AN=58;DP4=95,116,1,3;MQ=49	GT:PL:DP:AD:GP:GQ	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,15,140:5:5,0:0,26,169:26	0/0:0,27,253:9:9,0:0,38,282:38	0/0:0,3,40:1:1,0:0,14,69:14	0/0:0,27,255:9:9,0:0,38,284:38	0/0:0,42,255:14:14,0:0,53,284:53	0/0:11,0,33:2:1,1:2,3,53:3	0/0:0,3,29:1:1,0:0,14,58:14	0/0:0,3,17:1:1,0:0,14,46:14	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,3,29:1:1,0:0,14,58:14	0/0:0,27,255:9:9,0:0,38,284:38	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,3,41:1:1,0:0,14,70:14	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,12,140:4:4,0:0,23,169:23	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,3,33:1:1,0:0,14,62:14	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,6,81:2:2,0:0,17,110:17	0/0:0,12,124:4:4,0:0,23,153:23	0/0:0,3,39:1:1,0:0,14,68:14	0/0:0,27,215:9:9,0:0,38,244:38	0/1:58,0,32:3:1,2:46,0,49:44	0/0:0,12,143:4:4,0:0,23,172:23	0/0:0,9,109:3:3,0:0,20,138:20	0/0:0,172,255:57:57,0:0,183,284:127	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,18,180:6:6,0:0,29,209:29	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,5,97:4:3,1:0,16,126:16	0/0:0,21,214:7:7,0:0,32,243:32	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,78,255:26:26,0:0,89,284:89	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,30,254:10:10,0:0,41,283:41	0/0:0,12,136:4:4,0:0,23,165:23	0/0:0,9,113:3:3,0:0,20,142:20	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,42,255:14:14,0:0,53,284:53	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0
OTAU017482-RA	857	.	G	A	14.3303	.	DP=244;VDB=0.732596;SGB=0.336893;RPB=0.818184;MQB=0.0120429;MQSB=0.971034;BQB=0.325725;MQ0F=0;ICB=0.00638978;HOB=0.00295858;AC=2;AN=52;DP4=160,59,3,0;MQ=49	GT:PL:DP:AD:GP:GQ	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,24,255:8:8,0:0,31,276:31	0/0:0,18,183:6:6,0:0,25,204:25	0/0:0,3,41:1:1,0:0,11,62:11	0/0:0,39,255:13:13,0:0,46,276:46	0/0:0,51,255:17:17,0:0,58,276:58	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:53,6,0:2:0,2:39,0,8:8	0/0:0,27,206:9:9,0:0,34,227:34	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:23,3,0:1:0,1:12,0,11:9	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,3,35:1:1,0:0,11,56:11	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,3,33:1:1,0:0,11,54:11	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,9,117:3:3,0:0,16,138:16	0/0:0,6,68:2:2,0:0,13,89:13	0/0:0,9,102:3:3,0:0,16,123:16	0/0:0,15,152:5:5,0:0,22,173:22	0/0:0,12,133:4:4,0:0,19,154:19	0/0:0,24,224:8:8,0:0,31,245:31	0/0:0,6,70:2:2,0:0,13,91:13	0/0:0,163,255:54:54,0:0,170,276:127	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,27,255:9:9,0:0,34,276:34	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,15,168:5:5,0:0,22,189:22	0/0:0,33,255:11:11,0:0,40,276:40	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,69,255:23:23,0:0,76,276:76	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,30,188:10:10,0:0,37,209:37	0/0:0,21,207:7:7,0:0,28,228:28	0/0:0,21,167:7:7,0:0,28,188:28	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,30,255:10:10,0:0,37,276:37	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0
OTAU017482-RA	875	.	C	T	19.0242	.	DP=248;VDB=0.89732;SGB=-0.118132;RPB=0.990534;MQB=0.00284808;MQSB=0.934507;BQB=0.839131;MQ0F=0;ICB=0.014069;HOB=0.00617284;AC=3;AN=54;DP4=136,75,3,1;MQ=49	GT:PL:DP:AD:GP:GQ	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,21,221:7:7,0:0,27,239:27	0/0:0,15,164:5:5,0:0,21,182:21	0/0:0,6,78:2:2,0:0,12,96:12	0/0:0,36,255:12:12,0:0,42,273:42	0/0:0,36,255:12:12,0:0,42,273:42	0/1:18,3,0:1:0,1:9,0,10:6	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:53,6,0:2:0,2:41,0,7:7	0/0:0,21,213:7:7,0:0,27,231:27	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:23,3,0:1:0,1:14,0,9:8	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,3,35:1:1,0:0,9,53:9	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,3,35:1:1,0:0,9,53:9	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,9,109:3:3,0:0,15,127:15	0/0:0,9,91:3:3,0:0,15,109:15	0/0:0,9,119:3:3,0:0,15,137:15	0/0:0,18,151:6:6,0:0,24,169:24	0/0:0,9,113:3:3,0:0,15,131:15	0/0:0,27,254:9:9,0:0,33,272:33	0/0:0,6,70:2:2,0:0,12,88:12	0/0:0,141,255:47:47,0:0,147,273:127	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,27,255:9:9,0:0,33,273:33	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,21,234:7:7,0:0,27,252:27	0/0:0,39,255:13:13,0:0,45,273:45	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,75,255:25:25,0:0,81,273:81	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,39,255:13:13,0:0,45,273:45	0/0:0,21,238:7:7,0:0,27,256:27	0/0:0,21,172:7:7,0:0,27,190:27	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,21,203:7:7,0:0,27,221:27	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0
OTAU017482-RA	1024	.	A	T	89	.	DP=241;VDB=0.127985;SGB=11.0361;RPB=0.941479;MQB=0.00360482;MQSB=0.998372;BQB=0.959379;MQ0F=0;ICB=0.00193141;HOB=0.00094518;AC=1;AN=46;DP4=140,62,3,1;MQ=49	GT:PL:DP:AD:GP:GQ	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,30,255:10:10,0:0,40,281:40	0/0:0,18,197:6:6,0:0,28,223:28	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,45,255:15:15,0:0,55,281:55	0/0:0,33,255:11:11,0:0,43,281:43	0/1:140,12,0:4:0,4:118,1,5:5	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,12,132:4:4,0:0,22,158:22	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,3,35:1:1,0:0,13,62:13	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,12,145:4:4,0:0,22,171:22	0/0:0,3,29:1:1,0:0,13,56:13	0/0:0,18,202:6:6,0:0,28,228:28	0/0:0,24,231:8:8,0:0,34,257:34	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,12,144:4:4,0:0,22,170:22	0/0:0,12,140:4:4,0:0,22,166:22	0/0:0,148,255:49:49,0:0,158,281:127	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,30,255:10:10,0:0,40,281:40	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,15,147:5:5,0:0,25,173:25	0/0:0,15,164:5:5,0:0,25,190:25	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,45,255:15:15,0:0,55,281:55	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,48,255:16:16,0:0,58,281:58	0/0:0,18,185:6:6,0:0,28,211:28	0/0:0,24,194:8:8,0:0,34,220:34	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,3,35:1:1,0:0,13,62:13	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,39,255:13:13,0:0,49,281:49	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0
OTAU017482-RA	1103	.	C	T	62	.	DP=244;VDB=0.740842;SGB=11.8527;RPB=0.929623;MQB=0.0061008;MQSB=0.999998;BQB=0.979407;MQ0F=0;ICB=0.00163197;HOB=0.0008;AC=1;AN=50;DP4=111,120,0,4;MQ=49	GT:PL:DP:AD:GP:GQ	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,27,251:9:9,0:0,37,278:37	0/0:0,24,240:8:8,0:0,34,267:34	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,42,255:14:14,0:0,52,282:52	0/0:0,42,255:14:14,0:0,52,282:52	0/1:114,12,0:4:0,4:92,1,6:6	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,24,237:8:8,0:0,34,264:34	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,6,90:2:2,0:0,16,117:16	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,12,121:4:4,0:0,22,148:22	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,18,182:6:6,0:0,28,209:28	0/0:0,6,74:2:2,0:0,16,101:16	0/0:0,27,255:9:9,0:0,37,282:37	0/0:0,33,255:11:11,0:0,43,282:43	0/0:0,12,112:4:4,0:0,22,139:22	0/0:0,24,222:8:8,0:0,34,249:34	0/0:0,6,91:2:2,0:0,16,118:16	0/0:0,154,255:51:51,0:0,164,282:127	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,33,255:11:11,0:0,43,282:43	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,12,139:4:4,0:0,22,166:22	0/0:0,21,207:7:7,0:0,31,234:31	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,63,255:21:21,0:0,73,282:73	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,51,255:17:17,0:0,61,282:61	0/0:0,12,120:4:4,0:0,22,147:22	0/0:0,18,190:6:6,0:0,28,217:28	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,3,36:1:1,0:0,13,63:13	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,24,241:8:8,0:0,34,268:34	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0
OTAU017482-RA	1136	.	T	C	521	.	DP=247;VDB=0.0192071;SGB=53.3482;RPB=0.937648;MQB=1.94112e-13;MQSB=0.950755;BQB=0.174714;MQ0F=0;ICB=0.295805;HOB=0.135417;AC=6;AN=48;DP4=96,98,14,11;MQ=49	GT:PL:DP:AD:GP:GQ	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,24,221:8:8,0:0,27,234:27	1/1:248,27,0:9:0,9:234,17,0:17	./.:0,0,0:0:0,0:0,0,0:0	1/1:255,36,0:12:0,12:241,26,0:26	0/0:0,36,255:12:12,0:0,39,268:39	0/1:58,6,0:2:0,2:49,1,5:5	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,15,149:5:5,0:0,19,163:19	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:63,6,0:2:0,2:54,1,5:5	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,27,255:9:9,0:0,30,268:30	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,12,139:4:4,0:0,16,153:16	0/0:0,9,97:3:3,0:0,13,111:13	0/0:0,21,231:7:7,0:0,24,244:24	0/0:0,30,255:10:10,0:0,33,268:33	0/0:0,12,129:4:4,0:0,16,143:16	0/0:0,24,250:8:8,0:0,27,263:27	0/0:0,6,64:2:2,0:0,10,78:10	0/0:0,129,255:43:43,0:0,132,268:127	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,21,209:7:7,0:0,24,222:24	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,9,98:3:3,0:0,13,112:13	0/0:0,9,98:3:3,0:0,13,112:13	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,93,255:31:31,0:0,96,268:96	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,51,255:17:17,0:0,54,268:54	0/0:0,3,34:1:1,0:0,7,48:7	0/0:0,15,162:5:5,0:0,19,176:19	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,36,255:12:12,0:0,39,268:39	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0
OTAU017482-RA	1168	.	T	C	11.62	.	DP=248;VDB=0.1;SGB=3.41636;RPB=0.62619;MQB=0;MQSB=0.912094;BQB=0.561905;MQ0F=0;ICB=0.00177225;HOB=0.000868056;AC=1;AN=48;DP4=102,108,1,1;MQ=49	GT:PL:DP:AD:GP:GQ	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,24,236:8:8,0:0,34,263:34	0/0:0,42,255:14:14,0:0,52,282:52	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,36,255:12:12,0:0,46,282:46	0/0:0,30,255:10:10,0:0,40,282:40	0/1:58,6,0:2:0,2:41,0,10:10	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,9,110:3:3,0:0,19,137:19	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,6,73:2:2,0:0,16,100:16	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,30,255:10:10,0:0,40,282:40	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,12,146:4:4,0:0,22,173:22	0/0:0,9,94:3:3,0:0,19,121:19	0/0:0,15,173:5:5,0:0,25,200:25	0/0:0,24,255:8:8,0:0,34,282:34	0/0:0,15,192:5:5,0:0,25,219:25	0/0:0,27,253:9:9,0:0,37,280:37	0/0:0,9,110:3:3,0:0,19,137:19	0/0:0,138,255:46:46,0:0,148,282:127	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,18,224:6:6,0:0,28,251:28	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,9,111:3:3,0:0,19,138:19	0/0:0,9,102:3:3,0:0,19,129:19	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,75,255:25:25,0:0,85,282:85	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,48,255:16:16,0:0,58,282:58	0/0:0,6,77:2:2,0:0,16,104:16	0/0:0,6,75:2:2,0:0,16,102:16	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,33,255:11:11,0:0,43,282:43	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0
OTAU017482-RA	1390	.	G	A	215	.	DP=248;VDB=9.00613e-08;SGB=-23.057;RPB=8.40628e-07;MQB=1.01764e-09;MQSB=0.423916;BQB=0.263757;MQ0F=0;ICB=0.653627;HOB=0.0730969;AC=13;AN=68;DP4=176,32,8,7;MQ=48	GT:PL:DP:AD:GP:GQ	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,15,142:5:5,0:0,15,149:15	0/1:17,0,89:4:3,1:16,0,95:16	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,33,255:11:11,0:0,33,262:33	0/0:0,33,238:11:11,0:0,33,245:33	0/0:0,3,34:1:1,0:1,5,42:5	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,30,255:10:10,0:0,30,262:30	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:29,0,29:2:1,1:28,0,35:2./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,6,71:2:2,0:0,7,79:7	./.:0,0,0:0:0,0:0,0,0:0	0/1:23,3,0:1:0,1:20,1,5:5	0/1:23,0,35:2:1,1:22,0,41:22	0/1:34,3,0:1:0,1:31,1,5:5	0/0:0,18,170:6:6,0:0,18,177:18	0/0:0,18,188:6:6,0:0,18,195:18	0/0:0,6,70:2:2,0:0,7,78:7	0/0:0,48,255:16:16,0:0,48,262:48	0/0:0,36,209:12:12,0:0,36,216:36	0/0:0,9,110:3:3,0:0,10,117:10	0/0:0,12,124:4:4,0:0,12,131:12	0/0:0,132,255:44:44,0:0,132,262:127	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,18,163:6:6,0:0,18,170:18	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,12,130:4:4,0:0,12,137:12	0/1:15,0,142:7:5,2:14,0,148:14	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,63,255:21:21,0:0,63,262:63	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,39,255:13:13,0:0,39,262:39	0/0:0,9,123:3:3,0:0,10,130:10	0/0:0,21,209:7:7,0:0,21,216:21	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:29,3,0:1:0,1:26,1,5:5	0/1:17,3,0:1:0,1:15,1,5:4	0/1:29,3,0:1:0,1:26,1,5:5	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:58,6,0:2:0,2:54,2,3:3	0/0:0,33,255:11:11,0:0,33,262:33	0/1:29,3,0:1:0,1:26,1,5:5	0/1:29,3,0:1:0,1:26,1,5:5	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:29,3,0:1:0,1:26,1,5:5	./.:0,0,0:0:0,0:0,0,0:0
OTAU017482-RA	1398	.	A	G	225	.	DP=248;VDB=1.08313e-07;SGB=-22.0314;RPB=2.27216e-05;MQB=2.76382e-10;MQSB=0.546358;BQB=0.147491;MQ0F=0;ICB=0.653627;HOB=0.0730969;AC=13;AN=68;DP4=166,43,8,8;MQ=48	GT:PL:DP:AD:GP:GQ	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,15,140:5:5,0:0,15,147:15	0/1:11,0,151:6:5,1:10,0,158:10	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,30,255:10:10,0:0,30,262:30	0/0:0,36,239:12:12,0:0,36,246:36	0/0:0,3,36:1:1,0:1,5,44:5	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,27,250:9:9,0:0,27,257:27	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:28,0,30:2:1,1:27,0,36:26	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,6,71:2:2,0:0,7,79:7	./.:0,0,0:0:0,0:0,0,0:0	0/1:23,3,0:1:0,1:20,1,5:5	0/1:23,0,35:2:1,1:22,0,41:22	0/1:36,3,0:1:0,1:33,1,5:5	0/0:0,18,172:6:6,0:0,18,179:18	0/0:0,18,187:6:6,0:0,18,194:18	0/0:0,9,108:3:3,0:0,10,115:10	0/0:0,48,255:16:16,0:0,48,262:48	0/0:0,36,206:12:12,0:0,36,213:36	0/0:0,9,110:3:3,0:0,10,117:10	0/0:0,12,121:4:4,0:0,12,128:12	0/0:0,135,255:45:45,0:0,135,262:127	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,18,182:6:6,0:0,18,189:18	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,12,139:4:4,0:0,12,146:12	0/1:15,0,151:7:5,2:14,0,157:14	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,60,255:20:20,0:0,60,262:60	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,42,255:14:14,0:0,42,262:42	0/0:0,9,124:3:3,0:0,10,131:10	0/0:0,21,214:7:7,0:0,21,221:21	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:29,3,0:1:0,1:26,1,5:5	0/1:17,3,0:1:0,1:14,1,5:4	0/1:29,3,0:1:0,1:26,1,5:5	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:58,6,0:2:0,2:53,2,3:3	0/0:0,27,255:9:9,0:0,27,262:27	0/1:29,3,0:1:0,1:26,1,5:5	0/1:29,3,0:1:0,1:26,1,5:5	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:46,6,0:2:0,2:41,2,3:3	./.:0,0,0:0:0,0:0,0,0:0
OTAU017483-RA	768	.	A	G	246	.	DP=226;VDB=0.962691;SGB=22.1453;RPB=0.577251;MQB=1;MQSB=1;BQB=0.0196293;MQ0F=0.845133;ICB=0.298732;HOB=0.0897741;AC=98;AN=116;DP4=24,28,76,63;MQ=0	GT:PL:DP:AD:GP:GQ	1/1:8,6,0:2:0,2:15,7,0:6	0/1:0,3,4:1:1,0:7,3,4:2	1/1:8,6,0:2:0,2:15,7,0:6	1/1:0,1,0:2:1,1:9,4,2:3	1/1:4,3,0:1:0,1:12,5,1:4	0/1:0,3,4:1:1,0:7,3,4:2	./.:0,0,0:0:0,0:0,0,0:0	1/1:14,12,0:4:0,4:21,12,0:12	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	1/1:4,3,0:1:0,1:12,5,1:4	1/1:0,1,0:2:1,1:9,4,2:3	1/1:7,6,0:2:0,2:14,7,1:6	0/1:0,3,4:1:1,0:7,3,4:2	./.:0,0,0:0:0,0:0,0,0:0	1/1:4,3,0:1:0,1:12,5,1:4	1/1:8,6,0:2:0,2:15,7,0:6	1/1:7,6,0:2:0,2:14,7,1:6	0/0:0,9,11:3:3,0:2,5,6:3	1/1:4,3,0:1:0,1:12,5,1:4	./.:0,0,0:0:0,0:0,0,0:0	0/1:0,3,4:1:1,0:7,3,4:2	1/1:11,9,0:3:0,3:18,9,0:9	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	1/1:0,3,3:3:2,1:7,4,3:2	1/1:7,6,0:2:0,2:14,7,1:6	1/1:4,3,0:1:0,1:12,5,1:4	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	0/1:0,3,4:1:1,0:7,3,4:2	0/1:0,3,4:1:1,0:7,3,4:2	./.:0,0,0:0:0,0:0,0,0:0	0/0:0,9,11:3:3,0:2,5,6:3	1/1:8,6,0:2:0,2:15,7,0:6	1/1:4,3,0:1:0,1:12,5,1:4	1/1:0,4,1:7:4,3:8,5,2:3	1/1:16,15,0:5:0,5:23,15,0:14	1/1:12,18,0:6:0,6:19,18,0:15	1/1:4,3,0:1:0,1:12,5,1:4	1/1:14,12,0:4:0,4:21,12,0:12	1/1:23,30,0:10:0,10:29,30,0:27	1/1:4,3,0:1:0,1:12,5,1:4	1/1:1,3,0:7:3,4:10,5,2:4	1/1:0,1,0:2:1,1:9,4,2:3	1/1:7,6,0:2:0,2:14,7,1:6	1/1:12,15,0:5:0,5:19,15,0:14	1/1:17,15,0:5:0,5:24,15,0:15	0/1:0,3,4:1:1,0:7,3,4:2	1/1:11,9,0:3:0,3:18,9,0:9	./.:0,0,0:0:0,0:0,0,0:0	1/1:4,3,0:1:0,1:12,5,1:4	0/1:0,3,4:1:1,0:7,3,4:2	0/1:0,6,7:2:2,0:4,4,4:1	0/1:0,3,4:1:1,0:7,3,4:2	./.:0,0,0:0:0,0:0,0,0:0	./.:0,0,0:0:0,0:0,0,0:0	1/1:4,3,0:1:0,1:12,5,1:4	1/1:4,3,0:1:0,1:12,5,1:4	./.:0,0,0:0:0,0:0,0,0:0	1/1:14,15,0:5:0,5:21,15,0:14	1/1:14,12,0:4:0,4:21,12,0:12	0/0:0,7,8:7:5,2:4,4,5:2	1/1:8,6,0:2:0,2:15,7,0:6	1/1:0,3,3:5:3,2:7,4,3:2	1/1:17,15,0:5:0,5:24,15,0:15	1/1:5,5,0:6:2,4:13,6,1:5	1/1:0,3,0:12:6,6:9,5,2:4	1/1:23,27,0:9:0,9:29,27,0:25	1/1:20,24,0:8:0,8:26,24,0:22	1/1:20,18,0:6:0,6:26,18,0:17	0/0:0,9,8:10:7,3:3,6,4:2
[lerogers@pbio381 snps]$ vi OTAU_2018_reads2snps.vcf
[lerogers@pbio381 snps]$ pwd
/data/project_data/beetles/snps
[lerogers@pbio381 snps]$ cd ~
[lerogers@pbio381 ~]$ cd myresults
[lerogers@pbio381 myresults]$ vcftools

VCFtools (0.1.14)
© Adam Auton and Anthony Marcketta 2009

Process Variant Call Format files

For a list of options, please go to:
	https://vcftools.github.io/man_latest.html

Alternatively, a man page is available, type:
	man vcftools

Questions, comments, and suggestions should be emailed to:
	vcftools-help@lists.sourceforge.net

[lerogers@pbio381 myresults]$ vcftools --vcf /data/project_data/beetles/snps

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf /data/project_data/beetles/snps

Error: Does not appear to be a regular file: /data/project_data/beetles/snps
[lerogers@pbio381 myresults]$ vcftools --vcf /data/project_data/beetles/snps/OTAU_2018_samtools.vcf 

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf /data/project_data/beetles/snps/OTAU_2018_samtools.vcf

After filtering, kept 72 out of 72 Individuals
After filtering, kept 611928 out of a possible 611928 Sites
Run Time = 3.00 seconds
[lerogers@pbio381 myresults]$ ll
total 4
-rw-r--r--. 1 lerogers users 283 Feb 28 11:21 out.log
[lerogers@pbio381 myresults]$ vcftools --vcf /data/project_data/beetles/snps/OTAU_2018_samtools.vcf --min-alleles 2 --max-alleles 2

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf /data/project_data/beetles/snps/OTAU_2018_samtools.vcf
	--max-alleles 2
	--min-alleles 2

After filtering, kept 72 out of 72 Individuals
After filtering, kept 603231 out of a possible 611928 Sites
Run Time = 4.00 seconds
[lerogers@pbio381 myresults]$ vcftools --vcf /data/project_data/beetles/snps/OTAU_2018_samtools.vcf --min-alleles 2 --max-alleles 2 --minGQ 10

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf /data/project_data/beetles/snps/OTAU_2018_samtools.vcf
	--max-alleles 2
	--min-alleles 2
	--minGQ 10

After filtering, kept 72 out of 72 Individuals
After filtering, kept 603231 out of a possible 611928 Sites
Run Time = 15.00 seconds
[lerogers@pbio381 myresults]$ vcftools --vcf /data/project_data/beetles/snps/OTAU_2018_samtools.vcf --min-alleles 2 --max-alleles 2 --minGQ 10 --max-missing 0.8

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf /data/project_data/beetles/snps/OTAU_2018_samtools.vcf
	--max-alleles 2
	--min-alleles 2
	--minGQ 10
	--max-missing 0.8

After filtering, kept 72 out of 72 Individuals
After filtering, kept 113284 out of a possible 611928 Sites
Run Time = 21.00 seconds
[lerogers@pbio381 myresults]$ vcftools --vcf /data/project_data/beetles/snps/OTAU_2018_samtools.vcf --min-alleles 2 --max-alleles 2 --minGQ 10 --max-missing 0.8 --maf 0.01

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf /data/project_data/beetles/snps/OTAU_2018_samtools.vcf
	--maf 0.01
	--max-alleles 2
	--min-alleles 2
	--minGQ 10
	--max-missing 0.8

After filtering, kept 72 out of 72 Individuals
After filtering, kept 83120 out of a possible 611928 Sites
Run Time = 20.00 seconds
[lerogers@pbio381 myresults]$ vcftools --vcf /data/project_data/beetles/snps/OTAU_2018_samtools.vcf --min-alleles 2 --max-alleles 2 --minGQ 10 --max-missing 0.8 --maf 0.01 --recode --out OTAU_2018_samtoolsFILTERED.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf /data/project_data/beetles/snps/OTAU_2018_samtools.vcf
	--maf 0.01
	--max-alleles 2
	--min-alleles 2
	--minGQ 10
	--max-missing 0.8
	--out OTAU_2018_samtoolsFILTERED.vcf
	--recode

After filtering, kept 72 out of 72 Individuals
Outputting VCF file...
After filtering, kept 83120 out of a possible 611928 Sites
Run Time = 38.00 seconds
[lerogers@pbio381 myresults]$ 



------
<div id='id-section12'/>
### Page 12:
------
<div id='id-section13'/>
### Page 13: 3-5-2018

From terminal, enter r using R command. Leave R using quit()

Save files to our myresults section of home directory, first one was for class comparison (see my paper notebook), second one was for what we decided to do together as a class, with --min-alleles 2, --max-alleles 2, --maf 0.01, --max-missing 0.8, and using reads2snps

Now we have a vcf file we are happy with, we can start doing population genomics with it: SFS (site frequency spectra - frequency distribution of snp alleles in population, tells pop history of rare vs common polymorphism)

cat = print to screen

cut -f cuts all but the "field" you tell it, ie the first column

Here is the terminal code:

Last login: Fri Mar  2 12:05:13 on ttys000
ip0af52de4:~ Lucy$ ssh lerogers@pbio381.uvm.edu
lerogers@pbio381.uvm.edu's password: 
Last login: Fri Mar  2 12:05:43 2018 from ip0af5239f.int.uvm.edu
[lerogers@pbio381 ~]$ cd /data/project_data/beetles/snps/
[lerogers@pbio381 snps]$ ll
total 11962612
drwxr-xr-x. 2 srkeller users          89 Mar  5 10:40 archive
-rw-r--r--. 1 srkeller users 10924248151 Mar  4 13:13 OTAU_2018_reads2snps_DP10GP95.vcf
-rw-r--r--. 1 srkeller users  1325465210 Feb 27 16:51 OTAU_2018_samtools.vcf
[lerogers@pbio381 snps]$ vcftools --vcf OTAU_2018_reads2snps_DP10GP95.vcf --min-alleles 2 --max-alleles 2 --maf 0.01 --min-GQ 20 --max-missing 1.0 --het --out ~/myresults/reads2snpsmiss1.0


Error: Unknown option: --min-GQ

[lerogers@pbio381 snps]$ vcftools --vcf OTAU_2018_reads2snps_DP10GP95.vcf --min-alleles 2 --max-alleles 2 --maf 0.01 --minGQ 20 --max-missing 1.0 --het --out ~/myresults/reads2snpsmiss1.0

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf OTAU_2018_reads2snps_DP10GP95.vcf
	--maf 0.01
	--max-alleles 2
	--min-alleles 2
	--minGQ 20
	--max-missing 1
	--het
	--out /users/l/e/lerogers/myresults/reads2snpsmiss1.0

After filtering, kept 72 out of 72 Individuals
Outputting Individual Heterozygosity
After filtering, kept 32723 out of a possible 33366006 Sites
Run Time = 96.00 seconds
[lerogers@pbio381 snps]$ vcftools --vcf OTAU_2018_reads2snps_DP10GP95.vcf --min-alleles 2 --max-alleles 2 --maf 0.01 --minGQ 10 --max-missing 1.0 --het --out ~/myresults/reads2snpsmiss1.0minGQ10

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf OTAU_2018_reads2snps_DP10GP95.vcf
	--maf 0.01
	--max-alleles 2
	--min-alleles 2
	--minGQ 10
	--max-missing 1
	--het
	--out /users/l/e/lerogers/myresults/reads2snpsmiss1.0minGQ10

After filtering, kept 72 out of 72 Individuals
Outputting Individual Heterozygosity
After filtering, kept 32723 out of a possible 33366006 Sites
Run Time = 90.00 seconds
[lerogers@pbio381 snps]$ cd ~
[lerogers@pbio381 ~]$ cd mydata/
[lerogers@pbio381 mydata]$ ll
total 12
-rw-r--r--. 1 lerogers users  720 Jan 24 11:30 cols_data_Fonly.txt
-rwxr-xr-x. 1 lerogers users 1474 Jan 24 11:25 cols_data.txt
drwxr-xr-x. 2 lerogers users 4096 Mar  2 13:04 sam
[lerogers@pbio381 mydata]$ cd ~
[lerogers@pbio381 ~]$ cd myresults/
[lerogers@pbio381 myresults]$ ll
total 173492
-rw-r--r--. 1 lerogers users       431 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.log
-rw-r--r--. 1 lerogers users 177630740 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       360 Feb 28 11:32 out.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:56 reads2snpsmiss1.0.het
-rw-r--r--. 1 lerogers users       438 Mar  5 10:56 reads2snpsmiss1.0.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:59 reads2snpsmiss1.0minGQ10.het
-rw-r--r--. 1 lerogers users       445 Mar  5 10:59 reads2snpsmiss1.0minGQ10.log
[lerogers@pbio381 myresults]$ 
[lerogers@pbio381 myresults]$ df <- read.table("reads2snpsmiss1.0.het",header=T)
-bash: syntax error near unexpected token `('
[lerogers@pbio381 myresults]$ R

R version 3.4.3 (2017-11-30) -- "Kite-Eating Tree"
Copyright (C) 2017 The R Foundation for Statistical Computing
Platform: x86_64-redhat-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> df <- read.table("reads2snpsmiss1.0.het",header=T)
> str(df)
'data.frame':	72 obs. of  5 variables:
 $ INDV   : Factor w/ 72 levels "IT_AD4_F1_","IT_AD4_F2_",..: 1 2 3 4 5 6 8 9 10 7 ...
 $ O.HOM. : int  25655 24939 25353 26213 24522 26079 25586 25269 24923 25712 ...
 $ E.HOM. : num  25524 25524 25524 25524 25524 ...
 $ N_SITES: int  32723 32723 32723 32723 32723 32723 32723 32723 32723 32723 ...
 $ F      : num  0.0182 -0.0813 -0.0238 0.0957 -0.1392 ...
> summary(df$F)
    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
-0.13919  0.01649  0.06757  0.05447  0.10022  0.19002 
> sd(df$F)
[1] 0.06923361
> quit()
Save workspace image? [y/n/c]: n
[lerogers@pbio381 myresults]$ cd /data/project_data/beetles/snps/
[lerogers@pbio381 snps]$ ll
total 11962612
drwxr-xr-x. 2 srkeller users          89 Mar  5 10:40 archive
-rw-r--r--. 1 srkeller users 10924248151 Mar  4 13:13 OTAU_2018_reads2snps_DP10GP95.vcf
-rw-r--r--. 1 srkeller users  1325465210 Feb 27 16:51 OTAU_2018_samtools.vcf
[lerogers@pbio381 snps]$ vcftools --vcf OTAU_2018_reads2snps_DP10GP95.vcf  --min-alleles 2 --max-alleles 2 --maf 0.01 --max-missing 0.8 --recode --out ~/myresults/OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf OTAU_2018_reads2snps_DP10GP95.vcf
	--maf 0.01
	--max-alleles 2
	--min-alleles 2
	--max-missing 0.8
	--out /users/l/e/lerogers/myresults/OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf
	--recode

After filtering, kept 72 out of 72 Individuals
Outputting VCF file...
After filtering, kept 133291 out of a possible 33366006 Sites
Run Time = 94.00 seconds
[lerogers@pbio381 snps]$ cd ~
[lerogers@pbio381 ~]$ cd myresults/
[lerogers@pbio381 myresults]$ ll
total 214560
-rw-r--r--. 1 lerogers users       458 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.log
-rw-r--r--. 1 lerogers users  42047182 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       431 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.log
-rw-r--r--. 1 lerogers users 177630740 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       360 Feb 28 11:32 out.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:56 reads2snpsmiss1.0.het
-rw-r--r--. 1 lerogers users       438 Mar  5 10:56 reads2snpsmiss1.0.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:59 reads2snpsmiss1.0minGQ10.het
-rw-r--r--. 1 lerogers users       445 Mar  5 10:59 reads2snpsmiss1.0minGQ10.log
[lerogers@pbio381 myresults]$ cat /data/project_data/beetles/metadata/cols_data.txt 
sample_ID	population	devstage	sex
IT_AD4_F1_	IT	AD4	F
IT_AD4_F2_	IT	AD4	F
IT_AD4_F3_	IT	AD4	F
IT_AD4_M1_	IT	AD4	M
IT_AD4_M2_	IT	AD4	M
IT_AD4_M3_	IT	AD4	M
IT_ADC_F13	IT	ADC	F
IT_ADC_F14	IT	ADC	F
IT_ADC_F15	IT	ADC	F
IT_ADC_F1_	IT	ADC	F
IT_ADC_F2_	IT	ADC	F
IT_ADC_F3_	IT	ADC	F
IT_ADC_F4_	IT	ADC	F
IT_ADC_F5_	IT	ADC	F
IT_ADC_F6_	IT	ADC	F
IT_ADC_M10	IT	ADC	M
IT_ADC_M11	IT	ADC	M
IT_ADC_M12	IT	ADC	M
IT_ADC_M16	IT	ADC	M
IT_ADC_M17	IT	ADC	M
IT_ADC_M18	IT	ADC	M
IT_ADC_M7_	IT	ADC	M
IT_ADC_M8_	IT	ADC	M
IT_ADC_M9_	IT	ADC	M
NC_AD4_F1_	NC	AD4	F
NC_AD4_F2_	NC	AD4	F
NC_AD4_F3_	NC	AD4	F
NC_AD4_M1_	NC	AD4	M
NC_AD4_M2_	NC	AD4	M
NC_AD4_M3_	NC	AD4	M
NC_L3L_F1_	NC	L3L	F
NC_L3L_F2_	NC	L3L	F
NC_L3L_F3_	NC	L3L	F
NC_L3L_M1_	NC	L3L	M
NC_L3L_M2_	NC	L3L	M
NC_L3L_M3_	NC	L3L	M
NC_PD1_F1_	NC	PD1	F
NC_PD1_F2_	NC	PD1	F
NC_PD1_F3_	NC	PD1	F
NC_PD1_M1_	NC	PD1	M
NC_PD1_M2_	NC	PD1	M
NC_PD1_M3_	NC	PD1	M
NC_PP1_F1_	NC	PP1	F
NC_PP1_F2_	NC	PP1	F
NC_PP1_F3_	NC	PP1	F
NC_PP1_M1_	NC	PP1	M
NC_PP1_M2_	NC	PP1	M
NC_PP1_M3_	NC	PP1	M
WA_AD4_F1_	WA	AD4	F
WA_AD4_F2_	WA	AD4	F
WA_AD4_F3_	WA	AD4	F
WA_AD4_M1_	WA	AD4	M
WA_AD4_M2_	WA	AD4	M
WA_AD4_M3_	WA	AD4	M
WA_L3L_F1_	WA	L3L	F
WA_L3L_F2_	WA	L3L	F
WA_L3L_F3_	WA	L3L	F
WA_L3L_M1_	WA	L3L	M
WA_L3L_M2_	WA	L3L	M
WA_L3L_M3_	WA	L3L	M
WA_PD1_F1_	WA	PD1	F
WA_PD1_F2_	WA	PD1	F
WA_PD1_F3_	WA	PD1	F
WA_PD1_M1_	WA	PD1	M
WA_PD1_M2_	WA	PD1	M
WA_PD1_M3_	WA	PD1	M
WA_PP1_F1_	WA	PP1	F
WA_PP1_F2_	WA	PP1	F
WA_PP1_F3_	WA	PP1	F
WA_PP1_M1_	WA	PP1	M
WA_PP1_M2_	WA	PP1	M
WA_PP1_M3_	WA	PP1	M
[lerogers@pbio381 myresults]$ grep "IT" /data/project_data/beetles/metadata/cols_data.txt 
IT_AD4_F1_	IT	AD4	F
IT_AD4_F2_	IT	AD4	F
IT_AD4_F3_	IT	AD4	F
IT_AD4_M1_	IT	AD4	M
IT_AD4_M2_	IT	AD4	M
IT_AD4_M3_	IT	AD4	M
IT_ADC_F13	IT	ADC	F
IT_ADC_F14	IT	ADC	F
IT_ADC_F15	IT	ADC	F
IT_ADC_F1_	IT	ADC	F
IT_ADC_F2_	IT	ADC	F
IT_ADC_F3_	IT	ADC	F
IT_ADC_F4_	IT	ADC	F
IT_ADC_F5_	IT	ADC	F
IT_ADC_F6_	IT	ADC	F
IT_ADC_M10	IT	ADC	M
IT_ADC_M11	IT	ADC	M
IT_ADC_M12	IT	ADC	M
IT_ADC_M16	IT	ADC	M
IT_ADC_M17	IT	ADC	M
IT_ADC_M18	IT	ADC	M
IT_ADC_M7_	IT	ADC	M
IT_ADC_M8_	IT	ADC	M
IT_ADC_M9_	IT	ADC	M
[lerogers@pbio381 myresults]$ grep "IT" /data/project_data/beetles/metadata/cols_data.txt | cut -f 1
IT_AD4_F1_
IT_AD4_F2_
IT_AD4_F3_
IT_AD4_M1_
IT_AD4_M2_
IT_AD4_M3_
IT_ADC_F13
IT_ADC_F14
IT_ADC_F15
IT_ADC_F1_
IT_ADC_F2_
IT_ADC_F3_
IT_ADC_F4_
IT_ADC_F5_
IT_ADC_F6_
IT_ADC_M10
IT_ADC_M11
IT_ADC_M12
IT_ADC_M16
IT_ADC_M17
IT_ADC_M18
IT_ADC_M7_
IT_ADC_M8_
IT_ADC_M9_
[lerogers@pbio381 myresults]$ grep "IT" /data/project_data/beetles/metadata/cols_data.txt | cut -f 1 >IT.inds
[lerogers@pbio381 myresults]$ ll
total 214564
-rw-r--r--. 1 lerogers users       264 Mar  5 11:31 IT.inds
-rw-r--r--. 1 lerogers users       458 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.log
-rw-r--r--. 1 lerogers users  42047182 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       431 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.log
-rw-r--r--. 1 lerogers users 177630740 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       360 Feb 28 11:32 out.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:56 reads2snpsmiss1.0.het
-rw-r--r--. 1 lerogers users       438 Mar  5 10:56 reads2snpsmiss1.0.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:59 reads2snpsmiss1.0minGQ10.het
-rw-r--r--. 1 lerogers users       445 Mar  5 10:59 reads2snpsmiss1.0minGQ10.log
[lerogers@pbio381 myresults]$ vcftools --vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf --keep IT.inds --freq2 --out IT

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
	--keep IT.inds
	--freq2
	--out IT

Keeping individuals in 'keep' list
After filtering, kept 24 out of 72 Individuals
Outputting Frequency Statistics...
After filtering, kept 133291 out of a possible 133291 Sites
Run Time = 2.00 seconds
[lerogers@pbio381 myresults]$ ll
total 219664
-rw-r--r--. 1 lerogers users   5218166 Mar  5 11:32 IT.frq
-rw-r--r--. 1 lerogers users       264 Mar  5 11:31 IT.inds
-rw-r--r--. 1 lerogers users       402 Mar  5 11:32 IT.log
-rw-r--r--. 1 lerogers users       458 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.log
-rw-r--r--. 1 lerogers users  42047182 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       431 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.log
-rw-r--r--. 1 lerogers users 177630740 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       360 Feb 28 11:32 out.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:56 reads2snpsmiss1.0.het
-rw-r--r--. 1 lerogers users       438 Mar  5 10:56 reads2snpsmiss1.0.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:59 reads2snpsmiss1.0minGQ10.het
-rw-r--r--. 1 lerogers users       445 Mar  5 10:59 reads2snpsmiss1.0minGQ10.log
[lerogers@pbio381 myresults]$ head IT.frq
CHROM	POS	N_ALLELES	N_CHR	{FREQ}
OTAU000033-RA	90	2	40	1	0
OTAU000033-RA	468	2	44	0.954545	0.0454545
OTAU000033-RA	510	2	46	0.956522	0.0434783
OTAU000033-RA	582	2	46	0.869565	0.130435
OTAU000033-RA	639	2	48	0.75	0.25
OTAU000033-RA	717	2	46	0.956522	0.0434783
OTAU000033-RA	726	2	46	1	0
OTAU000033-RA	738	2	44	1	0
OTAU000033-RA	753	2	46	1	0
[lerogers@pbio381 myresults]$ 


------
<div id='id-section14'/>
### Page 14: 3-7-2018

To do (on board at beginning of class)

1. create individual lists for each population (ex: group "IT" /data/project_data/beetles/metadata/cols.data.txt|cut -f 1)
2. Run vcftools to get allele frequencies for each population (Ex: vcftools -vcf filename.vcf --keep IT.inds --min....)
3. dowload .frq output files to your local drive for plotting in R

Last login: Tue Mar  6 13:35:14 on ttys000
ip0af52b85:~ Lucy$ ssh lerogers@pbio381.uvm.edu
lerogers@pbio381.uvm.edu's password: 
Last login: Mon Mar  5 10:39:54 2018 from ip0af52de4.int.uvm.edu
[lerogers@pbio381 ~]$ grep "NC" /data/project_data/beetles/metadata/cols_data.txt | cut -f 1 > NC.inds
[lerogers@pbio381 ~]$ grep "WA" /data/project_data/beetles/metadata/cols_data.txt | cut -f 1 > WA.inds
[lerogers@pbio381 ~]$ vcftools --vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf --keep NC.inds --freq2 --out IT

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
	--keep NC.inds
	--freq2
	--out IT

stat error: No such file or directory
Error: Can't determine file type of OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
[lerogers@pbio381 ~]$ ll
total 24
drwxr-xr-x. 2 lerogers users 4096 Jan 31 10:55 cleanreads
drwxr-xr-x. 2 lerogers users 4096 Jan 31 11:10 fastqcoutput
-rw-r--r--. 1 lerogers users  306 Mar  7 10:27 IT.log
drwxr-xr-x. 3 lerogers users   78 Feb 26 10:45 mydata
drwxr-xr-x. 2 lerogers users 4096 Mar  5 11:32 myresults
drwxr-xr-x. 2 lerogers users   82 Feb  7 10:30 myscripts
-rw-r--r--. 1 lerogers users  264 Mar  7 10:25 NC.inds
-rw-r--r--. 1 lerogers users  264 Mar  7 10:26 WA.inds
[lerogers@pbio381 ~]$ vcftools --vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf --keep NC.inds --freq2 --out NC

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
	--keep NC.inds
	--freq2
	--out NC

stat error: No such file or directory
Error: Can't determine file type of OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
[lerogers@pbio381 ~]$ cd myresults
[lerogers@pbio381 myresults]$ ll
total 219664
-rw-r--r--. 1 lerogers users   5218166 Mar  5 11:32 IT.frq
-rw-r--r--. 1 lerogers users       264 Mar  5 11:31 IT.inds
-rw-r--r--. 1 lerogers users       402 Mar  5 11:32 IT.log
-rw-r--r--. 1 lerogers users       458 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.log
-rw-r--r--. 1 lerogers users  42047182 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       431 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.log
-rw-r--r--. 1 lerogers users 177630740 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       360 Feb 28 11:32 out.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:56 reads2snpsmiss1.0.het
-rw-r--r--. 1 lerogers users       438 Mar  5 10:56 reads2snpsmiss1.0.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:59 reads2snpsmiss1.0minGQ10.het
-rw-r--r--. 1 lerogers users       445 Mar  5 10:59 reads2snpsmiss1.0minGQ10.log
[lerogers@pbio381 myresults]$ cp ~/NC.inds .
[lerogers@pbio381 myresults]$ cp ~/WA.inds .
[lerogers@pbio381 myresults]$ ll
total 219672
-rw-r--r--. 1 lerogers users   5218166 Mar  5 11:32 IT.frq
-rw-r--r--. 1 lerogers users       264 Mar  5 11:31 IT.inds
-rw-r--r--. 1 lerogers users       402 Mar  5 11:32 IT.log
-rw-r--r--. 1 lerogers users       264 Mar  7 10:31 NC.inds
-rw-r--r--. 1 lerogers users       458 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.log
-rw-r--r--. 1 lerogers users  42047182 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       431 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.log
-rw-r--r--. 1 lerogers users 177630740 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       360 Feb 28 11:32 out.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:56 reads2snpsmiss1.0.het
-rw-r--r--. 1 lerogers users       438 Mar  5 10:56 reads2snpsmiss1.0.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:59 reads2snpsmiss1.0minGQ10.het
-rw-r--r--. 1 lerogers users       445 Mar  5 10:59 reads2snpsmiss1.0minGQ10.log
-rw-r--r--. 1 lerogers users       264 Mar  7 10:31 WA.inds
[lerogers@pbio381 myresults]$ vcftools --vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf --keep NC.inds --freq2 --out NC

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
	--keep NC.inds
	--freq2
	--out NC

Keeping individuals in 'keep' list
After filtering, kept 24 out of 72 Individuals
Outputting Frequency Statistics...
After filtering, kept 133291 out of a possible 133291 Sites
Run Time = 2.00 seconds
[lerogers@pbio381 myresults]$ vcftools --vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf --keep WA.inds --freq2 --out WA

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
	--keep WA.inds
	--freq2
	--out WA

Keeping individuals in 'keep' list
After filtering, kept 24 out of 72 Individuals
Outputting Frequency Statistics...
After filtering, kept 133291 out of a possible 133291 Sites
Run Time = 2.00 seconds
[lerogers@pbio381 myresults]$ ll
total 229484
-rw-r--r--. 1 lerogers users   5218166 Mar  5 11:32 IT.frq
-rw-r--r--. 1 lerogers users       264 Mar  5 11:31 IT.inds
-rw-r--r--. 1 lerogers users       402 Mar  5 11:32 IT.log
-rw-r--r--. 1 lerogers users   4959806 Mar  7 10:31 NC.frq
-rw-r--r--. 1 lerogers users       264 Mar  7 10:31 NC.inds
-rw-r--r--. 1 lerogers users       402 Mar  7 10:31 NC.log
-rw-r--r--. 1 lerogers users       458 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.log
-rw-r--r--. 1 lerogers users  42047182 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       431 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.log
-rw-r--r--. 1 lerogers users 177630740 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       360 Feb 28 11:32 out.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:56 reads2snpsmiss1.0.het
-rw-r--r--. 1 lerogers users       438 Mar  5 10:56 reads2snpsmiss1.0.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:59 reads2snpsmiss1.0minGQ10.het
-rw-r--r--. 1 lerogers users       445 Mar  5 10:59 reads2snpsmiss1.0minGQ10.log
-rw-r--r--. 1 lerogers users   5077398 Mar  7 10:32 WA.frq
-rw-r--r--. 1 lerogers users       264 Mar  7 10:31 WA.inds
-rw-r--r--. 1 lerogers users       402 Mar  7 10:32 WA.log
[lerogers@pbio381 myresults]$ 


put on local machine: 

Last login: Wed Mar  7 10:20:44 on ttys000
ip0af52b85:~ Lucy$ cd /Users/Lucy/Desktop/Ecological\ Genomics\ Terminal\ Output\ files 
ip0af52b85:Ecological Genomics Terminal Output files Lucy$ scp lerogers@pbio381.uvm.edu:~/myresults/*frq .
lerogers@pbio381.uvm.edu's password: 
IT.frq                                        100% 5096KB   5.6MB/s   00:00    
NC.frq                                        100% 4844KB   4.6MB/s   00:01    
WA.frq                                        100% 4958KB   3.8MB/s   00:01    
ip0af52b85:Ecological Genomics Terminal Output files Lucy$ 



Then, in a new R script, we visualized the frequencies of SNP frequencies for each population

setwd("~/Desktop/Ecological Genomics Terminal Output files")

IT <- read.table("IT.frq",skip=1,header=F)
NC <- read.table("NC.frq",skip=1,header=F)
WA <- read.table("WA.frq",skip=1,header=F)

#head(IT)
#V1 = gene. V2 = location of SNP. V3 = # of alleles (2 by definition). v4 = number of copies (24 individuals = max of 28 copies). V5, V6 = frequency of each allele

IT <- hist(IT$V6, plot=F)
IT$counts <- IT$counts/sum(IT$counts)

NC <- hist(NC$V6, plot=F)
NC$counts <- NC$counts/sum(NC$counts)

WA <- hist(WA$V6, plot=F)
WA$counts <- WA$counts/sum(WA$counts)

plot(IT$counts, type="l", col="red", lwd=1.5,ylim=c(0,0.5))
lines(NC$counts, col="blue")
lines(WA$counts, col="green")





New workflow: vcf file --> (thin LD) --> cowef vcf .geno --> write bash script to run ADMIXTURE {K=1...10}-->R


Thin for LD: 

[lerogers@pbio381 myresults]$ vcftools --vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf  --thin 1000 --recode


Then we copied beetle.pop, beetle.spid, and bash script to our myscripts folder and used vim to open the beetle.spid text file and change the three output locations so that it goes to myscripts.

Then we opened the bash script. java -Xmx2G  indicates the # of GB of memory it's working with. If in the future you run out of memory, you can change that number.

We ran into issues running the script so we ended up copying a .geno file over from the public directory.



Then we opened up a new page with vim and started writing a bash script, called ADMIX.sh, which will run ADMIXTURE for k = 1...10. 

-rw-r--r--. 1 lerogers users       402 Mar  7 10:32 WA.log
[lerogers@pbio381 myresults]$ vim

#!/bin/bash

for K in {1..10}

do

admixture -j 4 --cv 10 ~/myresults/out.recode.vcf.geno $K | tee log${K}.out

done

grep "CV" log*out >chooseK.txt


~  



Here is the terminal part: 




Last login: Wed Mar  7 10:32:27 on ttys001
ip0af52b85:~ Lucy$ ssh lerogers@pbio381.uvm.edu
lerogers@pbio381.uvm.edu's password: 
Last login: Wed Mar  7 10:20:59 2018 from ip0af52b85.int.uvm.edu
[lerogers@pbio381 ~]$ cd myresults
[lerogers@pbio381 myresults]$ 
[lerogers@pbio381 myresults]$ ll
total 229484
-rw-r--r--. 1 lerogers users   5218166 Mar  5 11:32 IT.frq
-rw-r--r--. 1 lerogers users       264 Mar  5 11:31 IT.inds
-rw-r--r--. 1 lerogers users       402 Mar  5 11:32 IT.log
-rw-r--r--. 1 lerogers users   4959806 Mar  7 10:31 NC.frq
-rw-r--r--. 1 lerogers users       264 Mar  7 10:31 NC.inds
-rw-r--r--. 1 lerogers users       402 Mar  7 10:31 NC.log
-rw-r--r--. 1 lerogers users       458 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.log
-rw-r--r--. 1 lerogers users  42047182 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       431 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.log
-rw-r--r--. 1 lerogers users 177630740 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       360 Feb 28 11:32 out.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:56 reads2snpsmiss1.0.het
-rw-r--r--. 1 lerogers users       438 Mar  5 10:56 reads2snpsmiss1.0.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:59 reads2snpsmiss1.0minGQ10.het
-rw-r--r--. 1 lerogers users       445 Mar  5 10:59 reads2snpsmiss1.0minGQ10.log
-rw-r--r--. 1 lerogers users   5077398 Mar  7 10:32 WA.frq
-rw-r--r--. 1 lerogers users       264 Mar  7 10:31 WA.inds
-rw-r--r--. 1 lerogers users       402 Mar  7 10:32 WA.log
[lerogers@pbio381 myresults]$ vcftools --vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf  --thin 1000 --recode

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
	--thin 1000
	--recode

After filtering, kept 72 out of 72 Individuals
Outputting VCF file...
After filtering, kept 11964 out of a possible 133291 Sites
Run Time = 0.00 seconds
[lerogers@pbio381 myresults]$ cd ~/myscripts
[lerogers@pbio381 myscripts]$ cp /data/project_data/beetles/metadata/beetles.pop  .
[lerogers@pbio381 myscripts]$ cp /data/scripts/beetle.spid .
[lerogers@pbio381 myscripts]$ cp /data/scripts/vcf2geno.sh .
[lerogers@pbio381 myscripts]$ ll
total 28
-rwxr-xr-x. 1 lerogers users 1629 Mar  7 11:09 beetle.spid
-rw-r--r--. 1 lerogers users 1008 Mar  7 11:08 beetles.pop
-rwxr--r--. 1 lerogers users 6487 Feb  5 11:18 countxpression_PE.py
-rwxr--r--. 1 lerogers users   33 Feb  7 10:30 test_bash
-rwxr--r--. 1 lerogers users  764 Jan 31 10:49 trim_F3.sh
-rwxr-x--x. 1 lerogers users  233 Mar  7 11:09 vcf2geno.sh
[lerogers@pbio381 myscripts]$ head beetles.pop 
IT_AD4_F1_	IT
IT_AD4_F2_	IT
IT_AD4_F3_	IT
IT_AD4_M1_	IT
IT_AD4_M2_	IT
IT_AD4_M3_	IT
IT_ADC_F13	IT
IT_ADC_F14	IT
IT_ADC_F15	IT
IT_ADC_F1_	IT
[lerogers@pbio381 myscripts]$ head beetle.spid
# spid-file generated: Tue Oct 20 22:21:06 EDT 2015

# VCF Parser questions
PARSER_FORMAT=VCF

# Only output SNPs with a phred-scaled quality of at least:
VCF_PARSER_QUAL_QUESTION=
# Select population definition file:
VCF_PARSER_POP_FILE_QUESTION=~/mydata/beetle.pop
# What is the ploidy of the data?
[lerogers@pbio381 myscripts]$ vim beetle.spid
[lerogers@pbio381 myscripts]$ vim vcf2geno.sh 
[lerogers@pbio381 myscripts]$ vash vcf2geno.sh 
-bash: vash: command not found
[lerogers@pbio381 myscripts]$ bash vcf2geno.sh 
INFO  11:17:46 - load PGDSpider configuration from: /data/popgen/PGDSpider_2.0.9.0/spider.conf.xml
Warning: Could not get charToByteConverterClass!
initialize convert process...
read input file...

--------------------------------------------------------------------------------------
 PGDSpider 2.0.9.0 Copyright (c) 2007 - 2015 by Heidi Lischer, University of Bern

 Citation:
  Lischer HEL and Excoffier L (2012) PGDSpider: An automated data conversion tool for
  connecting population genetics and genomics programs. Bioinformatics 28: 298-299.
--------------------------------------------------------------------------------------

Usage: PGDSpiderCli <options...>

Options:
-? or -h               For this page.
-inputfile <file>      The input file for the conversion (mandatory).
-inputformat <format>  Format of the input file. Mandatory if not defined
                       in the answer (SPID) file.
                       Possible input formats:
                       PGD, ARLEQUIN, BAM, BAPS, BATWING, BCF, CONVERT, FASTA, FASTQ
                       FSTAT, GDA, GENELAND, GENEPOP, GENETIX, HGDP_CEPH, IMMANC,
                       IM, MEGA, MIGRATE, MSA, NEWHYBRIDS, NEXUS, PED, PHYLIP,
                       SAM, STRUCTURE, VCF
-outputfile <file>     The output file for the conversion (mandatory).
-outputformat <format> Format of the output file. Mandatory if not defined
                       in the answer (SPID) file.
                       Possible output formats:
                       PGD, ARLEQUIN, BAM, BAPS, BATWING, BCF, FASTA, FASTQ, FDIST2, FSTAT,
                       GDA, GENELAND, GENEPOP, GENETIX, GESTE_BAYE_SCAN, IMMANC, IM,
                       KML, MEGA, MIGRATE, MSA, MSVAR, NEWHYBRIDS, NEXUS, PED, PHYLIP,
                       SAM, STRUCTURE, VCF
-spid <file>           SPID file containing preanswered conversion questions
                       (mandatory).


java.lang.NullPointerException
java.lang.NullPointerException
	at ch.unibe.iee.cmpg.pgdspider.model.parser.VcfParser.parse(VcfParser.java:563)
	at ch.unibe.iee.cmpg.pgdspider.cli.CLIController.convert(CLIController.java:220)
	at ch.unibe.iee.cmpg.pgdspider.PGDSpiderCli.main(PGDSpiderCli.java:31)
[lerogers@pbio381 myscripts]$ vim vcf2geno.sh 
[lerogers@pbio381 myscripts]$ vim beetle.spid
[lerogers@pbio381 myscripts]$ vim vcf2geno.sh 
[lerogers@pbio381 myscripts]$ bash vcf2geno.sh 
INFO  11:21:04 - load PGDSpider configuration from: /data/popgen/PGDSpider_2.0.9.0/spider.conf.xml
Warning: Could not get charToByteConverterClass!
initialize convert process...
read input file...
ERROR 11:21:04 - File: ~/myscripts/beetles.pop not found!
read input file done.
write output file...
ERROR 11:21:06 - File: ~/myscripts/beetle not found!
ERROR 11:21:06 - File: ~/myscripts/beetle not found!
WARN  11:21:07 - Ind file not writable!
WARN  11:21:07 - SNP file not writable!
write output file done.
[lerogers@pbio381 myscripts]$ chmod u+x beetles.pop 
[lerogers@pbio381 myscripts]$ bash vcf2geno.sh 
INFO  11:21:32 - load PGDSpider configuration from: /data/popgen/PGDSpider_2.0.9.0/spider.conf.xml
Warning: Could not get charToByteConverterClass!
initialize convert process...
read input file...
ERROR 11:21:32 - File: ~/myscripts/beetles.pop not found!
read input file done.
write output file...
ERROR 11:21:35 - File: ~/myscripts/beetle not found!
ERROR 11:21:35 - File: ~/myscripts/beetle not found!
WARN  11:21:35 - Ind file not writable!
WARN  11:21:35 - SNP file not writable!
write output file done.
[lerogers@pbio381 myscripts]$ cd ~/myresults/
[lerogers@pbio381 myresults]$ cp /data/project_data/beetles/snps/*.geno .
[lerogers@pbio381 myresults]$ ll
total 233984
-rw-r--r--. 1 lerogers users   5218166 Mar  5 11:32 IT.frq
-rw-r--r--. 1 lerogers users       264 Mar  5 11:31 IT.inds
-rw-r--r--. 1 lerogers users       402 Mar  5 11:32 IT.log
-rw-r--r--. 1 lerogers users   4959806 Mar  7 10:31 NC.frq
-rw-r--r--. 1 lerogers users       264 Mar  7 10:31 NC.inds
-rw-r--r--. 1 lerogers users       402 Mar  7 10:31 NC.log
-rw-r--r--. 1 lerogers users       458 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.log
-rw-r--r--. 1 lerogers users  42047182 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       431 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.log
-rw-r--r--. 1 lerogers users 177630740 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.recode.vcf
-rw-r--r--. 1 lerogers users       342 Mar  7 11:02 out.log
-rw-r--r--. 1 lerogers users   3730115 Mar  7 11:02 out.recode.vcf
-rw-r--r--. 1 lerogers users    873372 Mar  7 11:25 out.recode.vcf.geno
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:56 reads2snpsmiss1.0.het
-rw-r--r--. 1 lerogers users       438 Mar  5 10:56 reads2snpsmiss1.0.log
-rw-r--r--. 1 lerogers users      2850 Mar  5 10:59 reads2snpsmiss1.0minGQ10.het
-rw-r--r--. 1 lerogers users       445 Mar  5 10:59 reads2snpsmiss1.0minGQ10.log
-rw-r--r--. 1 lerogers users   5077398 Mar  7 10:32 WA.frq
-rw-r--r--. 1 lerogers users       264 Mar  7 10:31 WA.inds
-rw-r--r--. 1 lerogers users       402 Mar  7 10:32 WA.log
[lerogers@pbio381 myresults]$ vim
[lerogers@pbio381 myresults]$ chmod u+x ADMIX.sh 
[lerogers@pbio381 myresults]$ 




------
<div id='id-section15'/>
### Page 15: 3-19-2018

take .spd file, vcf2geno.sh bash script, and beetle.pop file (associates each sample name with the population it belongs to), and copy all 3 to myscripts:

[lerogers@pbio381 myscripts]$ cp /data/project_data/beetles/metadata/beetle.pop .
[lerogers@pbio381 myscripts]$ cp /data/scripts/beetle.spid /data/scripts/vcf2geno.sh .

All 3 of these files need to be in the same directory, because that is the only place where this script can look, you can't put a pathway to another place. So now that we put everything in the same directory, we should be set to run the script.

We used vi to open the .spd file and checked the line labled #select population definition file. THIS IS WHAT WE CAN CHANGE IN THE FUTURE

Then we opened the bash script and looked at the java -Xmx2G which means to assign a max of 2G to the job. We can increase this value if we get an error telling us there isn't enough momory. The next flag, -jar, says it will give you a path to the directory, it won't change. The input file path WILL change...this is currently the path to the file where we thinned the SNPs down to only 1 per 1000 bp.  The output file has the same code as the input file but it has a different extention on it, the .geno extension. For output format, we could specify anything, but EIGENSOFT is what makes the .geno file. then we direct it to SPID, which is in myscripts.

Then, we read the script!  

[lerogers@pbio381 myscripts]$ bash vcf2geno.sh 

We checked myresults, and the file was successfully there, with today's date and time.

Then we peeked: 
[lerogers@pbio381 myresults]$ head out.recode.vcf.geno 

0 = AA
1 = AT
2 = TT
9 = missing

Next, we opened up our ADMIX script from our myresults folder and checked our loop that lets us run the loop for each value of K from 1 to 10. We took out the space between j and 4 and added an = between cv and 10, to make it work successfully.

For the test, low values of CV means it's good, there's a low degree of mismatch between the test and where it actually gets assigned. So the CV should be lowest for the best value of K.

The grep line will search for lines in the log files that contain the characters "CV" and then put them out into a new file called chooseK.txt.  The default value after grep is the whole line that contains the characters.

Now, we are checking the permissions of this script.

[lerogers@pbio381 myresults]$ ll ADMIX.sh 
-rwxr--r--. 1 lerogers users 151 Mar 19 10:43 ADMIX.sh

We have rwx permissions, so we're good.

Now, after running the script, in our myresuls file we have a .P file which contains allele frequencies, a .Q file which contains ancestry coefficients, and a .log logged output file, for each K value. We also have the single chooseK file. We can print that to screen: 

[lerogers@pbio381 myresults]$ cat chooseK.txt 
log10.out:CV error (K=10): 0.74078
log1.out:CV error (K=1): 0.44089
log2.out:CV error (K=2): 0.43635
log3.out:CV error (K=3): 0.44475
log4.out:CV error (K=4): 0.47987
log5.out:CV error (K=5): 0.51791
log6.out:CV error (K=6): 0.55534
log7.out:CV error (K=7): 0.58280
log8.out:CV error (K=8): 0.62599
log9.out:CV error (K=9): 0.66978

So, we are saying that none of these models are correct but now we know which ones might be useful to look at. So we will choose to look at K=2, 3, 4. So we need to download those Q files to an admixture file on our local machine, as well as bringing in the metadata (cols.txt)


******

Now, we are moving back to R in order create a new script called Population Structure OTAU.R


------
<div id='id-section16'/>
### Page 16: 03-21-2018

Discussion on board about calculating Fst

Prior odds of 10 is the default on BayeScan but it's a terrible idea, we should increase the prior to be more stringent. We typed in bayescan to terminal to get the default parameters that we can look at.

Now, we are writing a bash script to run the BayeScan Program. My prior odds are 5000. Then we did :w ~/myscripts/bayescan5000.sh to save it as a file with the name bayescan5000.sh in my myscripts folder. Then we gave ourselves permission to run it...from the myscripts folder we did 
[lerogers@pbio381 myscripts]$ chmod u+x bayescan5000.sh

We entered a screen and ran this script and will let it run over the weekend. ctrl A+D lets us exit the screen.  If we wanted to reattach, we would to screen -r.


------
<div id='id-section17'/>
### Page 17: 3-26-2018

We started off the coding part of class by talking about Bayesian Inference. 

Bayescan can be viewed as a 3D landscape where the height represents how likely the parameters are where x and y coordinates represent the parameters.

The Markov Chain can zero in on the heights on the landscape so you do more of the simulations there.

The simulator shows that you need to do multiple chains in order to have a higher probability of correctly understading a more complex landscape, such as one that has multiple peaks. 

Burn-in time: Time between when we start from a random spot and when we reach a stable sampling area.

In our data, the LL (log likelihood) is dependent on the allele frequencies for the high locus from each population, the amt of genetic drift that each pop has experienced, and the amount of natural selection that each locus underwent in addition to drift.

The first thing to do with MCMC is to look at the output, and that's really important. 

What do the output files contain? _AccRte.txt contains acceptance rate of output chain...how quickly it moves through likelihood space; fst.txt contains alpha values and posterior probability; .sel contains likelihood values throughout the chain; Verif.txt contains log data that tells you the parameters, etc

My run has not finished because my prior odds were high, at 5000...so, I am going to do this next part with one of Suraj's runs.  His data is named 10_fst.txt and 10.sel. We also need information on chromosome location of each SNP, since it isn't included in this part.  It is in the vcf file on our myresults folder. We ran the commans with vcftools and the file is out.kept.sites, and that has what we need.


Now we have 3 new files in our ecological genomics output files, and we are starting a new R script.

On Wednesday, we'll pick up on looking up which particular SNPS are under selection and what they do.


HEre is the command line from today: 



[lerogers@pbio381 myresults]$ ll *vcf
-rw-r--r--. 1 lerogers users  42047182 Mar  5 11:23 OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
-rw-r--r--. 1 lerogers users 177630740 Feb 28 11:34 OTAU_2018_samtoolsFILTERED.vcf.recode.vcf
-rw-r--r--. 1 lerogers users   3730115 Mar  7 11:02 out.recode.vcf
[lerogers@pbio381 myresults]$ vcftools --vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf --kept-sites 
 
VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf OTAU_2018_reads2snps_DP10GP95_biallelic_MAF01_Miss0.8.vcf.recode.vcf
	--kept-sites

After filtering, kept 72 out of 72 Individuals
Outputting Kept Sites...
After filtering, kept 133291 out of a possible 133291 Sites
Run Time = 2.00 seconds
[lerogers@pbio381 myresults]$ ll
total 254128




------
<div id='id-section18'/>
### Page 18:
------
<div id='id-section19'/>
### Page 19:
------
<div id='id-section20'/>
### Page 20:
------
<div id='id-section21'/>
### Page 21:
------
<div id='id-section22'/>
### Page 22:
------
<div id='id-section23'/>
### Page 23:
------
<div id='id-section24'/>
### Page 24:
------
<div id='id-section25'/>
### Page 25:
------
<div id='id-section26'/>
### Page 26:
------
<div id='id-section27'/>
### Page 27:
------
<div id='id-section28'/>
### Page 28:
------
<div id='id-section29'/>
### Page 29:
------
<div id='id-section30'/>
### Page 30:
------
<div id='id-section31'/>
### Page 31:
------
<div id='id-section32'/>
### Page 32:
------
<div id='id-section33'/>
### Page 33:
------
<div id='id-section34'/>
### Page 34:
------
<div id='id-section35'/>
### Page 35:
------
<div id='id-section36'/>
### Page 36:
------
<div id='id-section37'/>
### Page 37:
------
<div id='id-section38'/>
### Page 38:
------
<div id='id-section39'/>
### Page 39:
------
<div id='id-section40'/>
### Page 40:
------
<div id='id-section41'/>
### Page 41:
------
<div id='id-section42'/>
### Page 42:
------
<div id='id-section43'/>
### Page 43:
------
<div id='id-section44'/>
### Page 44:
------
<div id='id-section45'/>
### Page 45:
------
<div id='id-section46'/>
### Page 46:
------
<div id='id-section47'/>
### Page 47:
------
<div id='id-section48'/>
### Page 48:
------
<div id='id-section49'/>
### Page 49:
------
<div id='id-section50'/>
### Page 50:
------
<div id='id-section51'/>
### Page 51:
------
<div id='id-section52'/>
### Page 52:
------
<div id='id-section53'/>
### Page 53:
------
<div id='id-section54'/>
### Page 54:
------
<div id='id-section55'/>
### Page 55:
------
<div id='id-section56'/>
### Page 56:
------
<div id='id-section57'/>
### Page 57:
------
<div id='id-section58'/>
### Page 58:
------
<div id='id-section59'/>
### Page 59:
------
<div id='id-section60'/>
### Page 60:

------