## Author: Lucy Rogers     
## Ecological Genomics: Spring 2018

### Overall Description of notebook      

This notebook will catalog my entries throughout the semester, including all my code, results, and interpretations


### Date started: (2018-01-24)   
### Date end:   (year-month-day)    

### Philosophy   
Science should be reproducible and one of the best ways to achieve this is by logging research activities in a notebook. Because science/biology has increasingly become computational, it is easier to document computational projects in an electronic form, which can be shared online through Github.    

### Helpful features of the notebook     

**It is absolutely critical for your future self and others to follow your work.**     

* The notebook is set up with a series of internal links from the table of contents.    
* All notebooks should have a table of contents which has the "Page", date, and title (information that allows the reader to understand your work).     
* Also, one of the perks of keeping all activities in a single document is that you can **search and find elements quickly**.     
* Lastly, you can share specific entries because of the three "#" automatically creates a link when the notebook renders on github.      


<a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.  


### Table of contents for 60 entries (Format is *Page: Date(with year-month-day). Title*)        
* [Page 1: 2018-01-24](#id-section1). Intro to Github, RMarkdown, and UNIX command-line
* [Page 2: 2018-01-26](#id-section2). Independent practice with UNIX
* [Page 3: 2018-01-29](#id-section3). Visualize data by creating Fast QC report
* [Page 4:2018-01-31](#id-section4). Clean data using trimmomatic, visualize again, create sam directory and align sequences
* [Page 5:2018-02-05](#id-section5). Explore sam output, install DESeq into RStudio
* [Page 6:2018-02-07](#id-section6). Review bash script, use DESeq in R to write code to test for differences in gene expression
* [Page 7:2018-02-12](#id-section7). Writing R Script to test for differential gene expression
* [Page 8:2018-02-14](#id-section8). Writing R Script
* [Page 9:2018-02-21](#id-section9). Writing R Script, actual script is posted here
* [Page 10:](#id-section10).
* [Page 11:](#id-section11).
* [Page 12:](#id-section12).
* [Page 13:](#id-section13).
* [Page 14:](#id-section14).
* [Page 15:](#id-section15).
* [Page 16:](#id-section16).
* [Page 17:](#id-section17).
* [Page 18:](#id-section18).
* [Page 19:](#id-section19).
* [Page 20:](#id-section20).
* [Page 21:](#id-section21).
* [Page 22:](#id-section22).
* [Page 23:](#id-section23).
* [Page 24:](#id-section24).
* [Page 25:](#id-section25).
* [Page 26:](#id-section26).
* [Page 27:](#id-section27).
* [Page 28:](#id-section28).
* [Page 29:](#id-section29).
* [Page 30:](#id-section30).
* [Page 31:](#id-section31).
* [Page 32:](#id-section32).
* [Page 33:](#id-section33).
* [Page 34:](#id-section34).
* [Page 35:](#id-section35).
* [Page 36:](#id-section36).
* [Page 37:](#id-section37).
* [Page 38:](#id-section38).
* [Page 39:](#id-section39).
* [Page 40:](#id-section40).
* [Page 41:](#id-section41).
* [Page 42:](#id-section42).
* [Page 43:](#id-section43).
* [Page 44:](#id-section44).
* [Page 45:](#id-section45).
* [Page 46:](#id-section46).
* [Page 47:](#id-section47).
* [Page 48:](#id-section48).
* [Page 49:](#id-section49).
* [Page 50:](#id-section50).
* [Page 51:](#id-section51).
* [Page 52:](#id-section52).
* [Page 53:](#id-section53).
* [Page 54:](#id-section54).
* [Page 55:](#id-section55).
* [Page 56:](#id-section56).
* [Page 57:](#id-section57).
* [Page 58:](#id-section58).
* [Page 59:](#id-section59).
* [Page 60:](#id-section60).

------
<div id='id-section1'/>
### Page 1: 2018-01-24. Notes on using Github, Rmarkdown, and the UNIX command-line

Today, we created our Github repos for the course and began our notebooks.

Other goals for today:

* publish our notbooks to github
* log into the UNIX server


------
<div id='id-section2'/>
### Page 2: 2018-01-28 

Today, I reviewed the UNIX commands we learned on 2018-01-24. I reviewed the tutorial "Intro to connecting to unix servers and navigating the command-line" and changed the setting such that terminal will check with me before deleting a file via the rm function. Tomorrow in class, I will ask about why I cannot see my name linked under the class member page link on the course github page.

The most useful UNIX commands were

pwd = present working directory
mkdir = make directory
ll = long list
cd = change directory
cp filename path_to_destination = copy
head filename = print head...can also do with tail
grep 'word' filename = pull out all entries in file with the 'word' in them
grep 'IT' cols_data.txt >cols_ITonly.txt = save the ITonly output in the new file
man command = go to manual and tell me more about this command
wc = tally up lines, words, and characters in a dataset
rm filename = remove filename
grep 'F' filename | wc = use pipe to apply 2nd command to 1st command


------
<div id='id-section3'/>
### Page 3: 2018-01-29



Today, we learned about the bull headed beetle dataset and about the RNASeq data processing 'pipeline'.

First, we found our specific file within the group data file and used the zcat command to open it and the pipe command | to only show us the first 4 lines. We learned how to interpret this output data:
1. after @ is information about the reads
2. Then the actual sequence
3. sometimes more info
4. Quality data, the key is here http://www.drive5.com/usearch/manual/quality_score.html.


[lerogers@pbio381 ~]$ cd /data/project_data/beetles/rawdata/
[lerogers@pbio381 rawdata]$ zcat WA_PP1_F3_ATGTCA_L003_R1_001.fastq.gz  | head -n4
@CCRI0219:155:C2LNBACXX:3:1101:1226:1976 1:N:0:ATGTCA
TGGCGCCAAAAATAAGATGGAGAAAAAGAAATTAGTTGCGCATGCTGATGTTAAGTCTCGAGACCGGTACTATTTTAAATAAAAAAAATTTTATTTGATG
+
B@CFFFFFDDFFHIFGGGHGFHGIIIGIGHIIIJJDHIGGIGDHHGDCGIIEGE@=DHGIGGGHFDD>>?CDDEEDDDDEEDDCDDDDDCDDDDDDEDDC


Then we used the fastqc command to visualize our data and copied it into a new directory we made called fastqcoutput: 

[lerogers@pbio381 rawdata]$ cd ~
[lerogers@pbio381 ~]$ mkdir fastqcoutput
[lerogers@pbio381 ~]$ fastqc /data/project_data/beetles/rawdata/WA_PP1_F3_ATGTCA_L003_R1_001.fastq.gz -o ~/fastqcoutput
Started analysis of WA_PP1_F3_ATGTCA_L003_R1_001.fastq.gz
Approx 5% complete for WA_PP1_F3_ATGTCA_L003_R1_001.fastq.gz
...
Analysis complete for WA_PP1_F3_ATGTCA_L003_R1_001.fastq.gz


Using a second terminal window, we copied the fastqc output from the fastqcoutput folder on our home directory to a folder we created on our computers; mine is in the folder 'terminal output files' on my desktop. I did cd drag_file_symbol_here

scp = secure copy
 . = put it here

Terminal window 2:
ip0af52b37:~ Lucy$ cd /Users/Lucy/Desktop/Ecological\ Genomics\ Terminal\ Output\ files 
ip0af52b37:Ecological Genomics Terminal Output files Lucy$ pwd
/Users/Lucy/Desktop/Ecological Genomics Terminal Output files
ip0af52b37:Ecological Genomics Terminal Output files Lucy$ scp lerogers@pbio381.uvm.edu:~/fastqcoutput/* .
lerogers@pbio381.uvm.edu's password: 
WA_PP1_F3_ATGTCA_L003_R1_001_fastqc.html      100%  331KB 331.1KB/s   00:00    
WA_PP1_F3_ATGTCA_L003_R1_001_fastqc.zip       100%  428KB 428.0KB/s   00:00    
ip0af52b37:Ecological Genomics Terminal Output files Lucy$ 



------
<div id='id-section4'/>
### Page 4: 2018-01-31

We made a cleanreads folder on our home directory and then copied the example bash script from the group data folder to the myscripts folder on our home directory. We then opened the script using vim, which brought us to a new window on terminal. 


[lerogers@pbio381 ~]$ cd ~/
[lerogers@pbio381 ~]$ mkdir cleanreads
[lerogers@pbio381 ~]$ ll
total 4
drwxr-xr-x. 2 lerogers users    6 Jan 31 10:21 cleanreads
drwxr-xr-x. 2 lerogers users 4096 Jan 29 11:07 fastqcoutput
drwxr-xr-x. 2 lerogers users   64 Jan 24 11:30 mydata
drwxr-xr-x. 2 lerogers users    6 Jan 24 11:18 myscripts
[lerogers@pbio381 myscripts]$ cp /data/scripts/trim_example.sh ~/myscripts/
[lerogers@pbio381 myscripts]$ ll
total 4
-rwxr--r--. 1 lerogers users 780 Jan 31 10:27 trim_example.sh
[lerogers@pbio381 myscripts]$ vim trim_example.sh
[lerogers@pbio381 myscripts]$ ll
total 4
-rwxr--r--. 1 lerogers users 764 Jan 31 10:43 trim_example.sh
[lerogers@pbio381 myscripts]$ vim trim_example.sh
[lerogers@pbio381 myscripts]$ ll
total 8
-rwxr--r--. 1 lerogers users 764 Jan 31 10:43 trim_example.sh
-rw-r--r--. 1 lerogers users 764 Jan 31 10:48 trim_F3
[lerogers@pbio381 myscripts]$ ll
total 8
-rwxr--r--. 1 lerogers users 764 Jan 31 10:43 trim_example.sh
-rw-r--r--. 1 lerogers users 764 Jan 31 10:48 trim_F3
[lerogers@pbio381 myscripts]$ vim trim_example.sh
[lerogers@pbio381 myscripts]$ ll
total 12
-rwxr--r--. 1 lerogers users 764 Jan 31 10:43 trim_example.sh
-rw-r--r--. 1 lerogers users 764 Jan 31 10:48 trim_F3
-rw-r--r--. 1 lerogers users 764 Jan 31 10:49 trim_F3.sh
[lerogers@pbio381 myscripts]$ rm trim_F3
rm: remove regular file ‘trim_F3’? yes
[lerogers@pbio381 myscripts]$ ll
total 8
-rwxr--r--. 1 lerogers users 764 Jan 31 10:43 trim_example.sh
-rw-r--r--. 1 lerogers users 764 Jan 31 10:49 trim_F3.sh



Now, we ran the trimmomatic script by first using chmod, which allowed us to change the permissions so that the 'user' could 'execute' the script, then ran the command ./trim_F3.sh which told it to run. 

[lerogers@pbio381 myscripts]$ chmod u+x trim_F3.sh
[lerogers@pbio381 myscripts]$ ll
total 8
-rwxr--r--. 1 lerogers users 764 Jan 31 10:43 trim_example.sh
-rwxr--r--. 1 lerogers users 764 Jan 31 10:49 trim_F3.sh
[lerogers@pbio381 myscripts]$ vim trim_f3.sh
[lerogers@pbio381 myscripts]$ vim trim_F3.sh
[lerogers@pbio381 myscripts]$ rm trim_example.sh
rm: remove regular file ‘trim_example.sh’? yes
[lerogers@pbio381 myscripts]$ ll
total 4
-rwxr--r--. 1 lerogers users 764 Jan 31 10:49 trim_F3.sh
[lerogers@pbio381 myscripts]$ ./trim_F3.sh
TrimmomaticPE: Started with arguments: -threads 1 -phred33 /data/project_data/beetles/rawdata/WA_PP1_F3_ATGTCA_L003_R1_001.fastq.gz /data/project_data/beetles/rawdata/WA_PP1_F3_ATGTCA_L003_R2_001.fastq.gz /users/l/e/lerogers/cleanreads/WA_PP1_F3_R1_clean_paired.fa /users/l/e/lerogers/cleanreads/WA_PP1_F3_R1_clean_unpaired.fa /users/l/e/lerogers/cleanreads/WA_PP1_F3_R2_clean_paired.fa /users/l/e/lerogers/cleanreads/WA_PP1_F3_R2_clean_unpaired.fa ILLUMINACLIP:/data/popgen/Trimmomatic-0.33/adapters/TruSeq3-PE.fa:2:30:10 LEADING:28 TRAILING:28 SLIDINGWINDOW:6:28 HEADCROP:12 MINLEN:35
Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'
ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
Input Read Pairs: 16000000 Both Surviving: 11356978 (70.98%) Forward Only Surviving: 1897057 (11.86%) Reverse Only Surviving: 800823 (5.01%) Dropped: 1945142 (12.16%)
TrimmomaticPE: Completed successfully
[lerogers@pbio381 myscripts]$ cd ~/cleanreads/
[lerogers@pbio381 cleanreads]$ ll
total 5548308
-rw-r--r--. 1 lerogers users 2590374644 Jan 31 10:59 WA_PP1_F3_R1_clean_paired.fa
-rw-r--r--. 1 lerogers users  369021891 Jan 31 10:59 WA_PP1_F3_R1_clean_unpaired.fa
-rw-r--r--. 1 lerogers users 2562494874 Jan 31 10:59 WA_PP1_F3_R2_clean_paired.fa
-rw-r--r--. 1 lerogers users  159565225 Jan 31 10:59 WA_PP1_F3_R2_clean_unpaired.fa


Now, finally, we ran fastqc again to visualize after trimming, placing the final output in our fastqc file: 

[lerogers@pbio381 cleanreads]$ fastqc ~/cleanreads/WA_PP1_F3_R1_clean_paired.fa -o ~/fastqcoutput/
Started analysis of WA_PP1_F3_R1_clean_paired.fa
Approx 5% complete for WA_PP1_F3_R1_clean_paired.fa
...
Analysis complete for WA_PP1_F3_R1_clean_paired.fa
[lerogers@pbio381 cleanreads]$ cd ~/
[lerogers@pbio381 ~]$ ll
total 8
drwxr-xr-x. 2 lerogers users 4096 Jan 31 10:55 cleanreads
drwxr-xr-x. 2 lerogers users 4096 Jan 31 11:10 fastqcoutput
drwxr-xr-x. 2 lerogers users   64 Jan 24 11:30 mydata
drwxr-xr-x. 2 lerogers users   31 Jan 31 10:54 myscripts
[lerogers@pbio381 ~]$ cd fastqcoutput/
[lerogers@pbio381 fastqcoutput]$ ll
total 1384
-rw-r--r--. 1 lerogers users 339080 Jan 29 11:07 WA_PP1_F3_ATGTCA_L003_R1_001_fastqc.html
-rw-r--r--. 1 lerogers users 438237 Jan 29 11:07 WA_PP1_F3_ATGTCA_L003_R1_001_fastqc.zip
-rw-r--r--. 1 lerogers users 285373 Jan 31 11:10 WA_PP1_F3_R1_clean_paired.fa_fastqc.html
-rw-r--r--. 1 lerogers users 349105 Jan 31 11:10 WA_PP1_F3_R1_clean_paired.fa_fastqc.zip


And copied it to our computer in a second terminal window:

Second terminal window: 
Last login: Wed Jan 31 10:57:21 on ttys001
ip0af527d2:~ Lucy$ cd /Users/Lucy/Desktop/Ecological\ Genomics\ Terminal\ Output\ files 
ip0af527d2:Ecological Genomics Terminal Output files Lucy$ scp lerogers@pbio381.uvm.edu:~/fastqcoutput/*html .
lerogers@pbio381.uvm.edu's password: 
WA_PP1_F3_ATGTCA_L003_R1_001_fastqc.html      100%  331KB 331.1KB/s   00:00    
WA_PP1_F3_R1_clean_paired.fa_fastqc.html      100%  279KB 278.7KB/s   00:00    
ip0af527d2:Ecological Genomics Terminal Output files Lucy$ 


Back to first terminal window, we next mapped our reads to the reference transcriptome. It had already been previously downloaded to the directory from a website using the following commands: 
cd /data/project_data/beetles/reference/
wget https://i5k.nal.usda.gov/data/Arthropoda/onttau-%28Onthophagus_taurus%29/Current%20Genome%20Assembly/2.Official%20or%20Primary%20Gene%20Set/BCM_version_0.5.3/consensus_gene_set/OTAU.fna

Then we talked about how we will work within the program bwa to map the directory. The professors had previously indexed the reference transcriptome with the command $ bwa index /data/project_data/beetles/reference/OTAU.fna

Finally, we created a sam directory in our mydata directory. Then we opened up a screen and aligned our sequence that way, such that when we pressed screen -r, the program continued to run but our computer was detached from it. 

[lerogers@pbio381 fastqcoutput]$ cd ~
[lerogers@pbio381 ~]$ cd ~/mydata/
[lerogers@pbio381 mydata]$ mkdir sam
[lerogers@pbio381 mydata]$ ll
total 8
-rw-r--r--. 1 lerogers users  720 Jan 24 11:30 cols_data_Fonly.txt
-rwxr-xr-x. 1 lerogers users 1474 Jan 24 11:25 cols_data.txt
drwxr-xr-x. 2 lerogers users   10 Jan 31 11:28 sam
[lerogers@pbio381 mydata]$ cd sam
[lerogers@pbio381 sam]$ screen
[screen is terminating]
[lerogers@pbio381 sam]$ screen
[detached from 53787.pts-6.pbio381]
[lerogers@pbio381 sam]$ screen -r
[detached from 53787.pts-6.pbio381]
[lerogers@pbio381 sam]$ 



------
<div id='id-section5'/>
### Page 5: 2018-02-05

I am copying the review of the vim and screen tricks from the tutorial because I found them useful:

Review vim and screen tricks
vim is a text editor that can be used from the command line. It can be quite powerful for editing once you know a few tricks. We started with the basics: To open the file trim_myfile.sh

vim trim_myfile.sh 
To edit or “insert” text, type i

To exit the edit mode, hit esc

To save your changes, you “write” the file with :w . Note that you start all commands in vim with a colon. To quit the program, :q .

screen is another fantastically useful program that allows programs or scripts to run in the background if you “detach” from the screen. To open a screen window, type screen. Now you have your command prompt and can start a script or enter a command. To “detach”, type ctrl plus a plus d . To reattach, type  screen -r and the session ID you want to enter.



Then we explored our sam output, then tried to run the python script countexpression_PE and put the output as .txt in the myscripts folder, but it didn't work and I'm not exactly sure why.

Finally, we installed DESeq into RStudio but mine failed because I have to update my computer, R, and RStudio.

------
<div id='id-section6'/>
### Page 6: 2018-02-07

Bash script is a way to keep a record of what you do...I really should review bash, I'm still a bit unclear about it.


[lerogers@pbio381 ~]$ cd ~/myscripts/
[lerogers@pbio381 myscripts]$ ll
total 12
-rwxr--r--. 1 lerogers users 6487 Feb  5 11:18 countxpression_PE.py
-rwxr--r--. 1 lerogers users  764 Jan 31 10:49 trim_F3.sh
[lerogers@pbio381 myscripts]$ vim test_bash
[lerogers@pbio381 myscripts]$ ll
total 16
-rwxr--r--. 1 lerogers users 6487 Feb  5 11:18 countxpression_PE.py
-rw-r--r--. 1 lerogers users   33 Feb  7 10:30 test_bash
-rwxr--r--. 1 lerogers users  764 Jan 31 10:49 trim_F3.sh
[lerogers@pbio381 myscripts]$ chmod u+x test_bash 
[lerogers@pbio381 myscripts]$ ll
total 16
-rwxr--r--. 1 lerogers users 6487 Feb  5 11:18 countxpression_PE.py
-rwxr--r--. 1 lerogers users   33 Feb  7 10:30 test_bash
-rwxr--r--. 1 lerogers users  764 Jan 31 10:49 trim_F3.sh
[lerogers@pbio381 myscripts]$ ./test_bash 
hello world
[lerogers@pbio381 myscripts]$ 


Next, we got a new clean dataset that Melissa had prepared and we switched over to R, where we wrote the code to use DESeq to test for differences in gene expression. 

------
<div id='id-section7'/>
### Page 7: 2018-02-12

We worked on writing our R script for testing for differential gene expression

------
<div id='id-section8'/>
### Page 8: 2018-02-14
More R script work

------
<div id='id-section9'/>
### Page 9:2018-02-21

Finishing the R script. Here is the script, pasted below:

# set working directory
setwd("~/Desktop/Ecological Genomics Terminal Output files")

library("DESeq2")
library("ggplot2")

countsTable <- read.delim('allcountsdataRN_noIT.txt',header=TRUE, stringsAsFactors = TRUE, row.names = 1)
countData <- as.matrix(countsTable)
head(countData)

conds <- read.delim("cols_data_noIT.txt", header=TRUE, stringsAsFactors = TRUE, row.names = 1)
head(conds)
colData <- as.data.frame(conds)

####################################### we've now set everything up

dds <- DESeqDataSetFromMatrix(countData = countData, colData = colData, design = ~ devstage + sex + population)
# "population effect" model controlling for differences in devstage and sex. Population is listed last, so that's what it's ultimately testing for.

dim(dds)
#make note of output dimenstions: 17483, 48

dds <- dds[rowSums(counts(dds)) > 1,]

dim(dds)
#new dimensions: 16851, 48. Only 632 genes lost.

dds <- DESeq(dds, modelMatrixType = "standard")

resultsNames(dds)
#[1] "Intercept"           "devstage_L3L_vs_AD4" "devstage_PD1_vs_AD4"
#[4] "devstage_PP1_vs_AD4" "sex_M_vs_F"          "population_WA_vs_NC"

res <- results(dds)
str(res)

res <- res[order(res$padj),]
head(res)
#save output table here. use shift+command+C to put many lines as text

# log2 fold change (MLE): population WA vs NC
# Wald test p-value: population WA vs NC
# DataFrame with 6 rows and 6 columns
# baseMean log2FoldChange     lfcSE      stat       pvalue         padj
# <numeric>      <numeric> <numeric> <numeric>    <numeric>    <numeric>
#   OTAU017482-RA 126.291608     -5.4166340 0.6732174 -8.045892 8.561964e-16 1.268369e-11
# OTAU012716-RA 188.877675      4.2644034 0.5427535  7.856980 3.935069e-15 2.914706e-11
# OTAU008667-RA 231.871115     -0.8736955 0.1267134 -6.895051 5.384538e-12 1.994164e-08
# OTAU012562-RA 251.774364     -0.8774079 0.1270957 -6.903520 5.072966e-12 1.994164e-08
# OTAU013988-RA   4.416955      4.4229857 0.6836393  6.469765 9.815559e-11 2.908154e-07
# OTAU011160-RA  10.241516     -2.5149390 0.4125305 -6.096371 1.085033e-09 2.678947e-06

summary(res)

# out of 16851 with nonzero total read count
# adjusted p-value < 0.1
# LFC > 0 (up)     : 333, 2% 
# LFC < 0 (down)   : 282, 1.7% 
# outliers [1]     : 85, 0.5% 
# low counts [2]   : 1952, 12% 
# (mean count < 1)
# [1] see 'cooksCutoff' argument of ?results
# [2] see 'independentFiltering' argument of ?results

#FOR NEXT TIME: what model would you want to do, and how would you set it up???

res_pop <- results(dds, name="population_WA_vs_NC", alpha=0.05)

res_pop <- res_pop[order(res_pop$padj),]
summary(res_pop)

#################### Data Visualization
plotMA(res_pop, main="DESeq2", ylim=c(-2,2))
abline(h=c(-1,1), col="blue", lwd=2)

# sex effect?

res_sex <- results(dds, name="sex_M_vs_F", alpha=0.05)
plotMA(res_sex, main="DESeq2", ylim=c(-4,4))
abline(h=c(-1,1), col="blue", lwd=2)

###### PCA
rld <- rlog(dds, blind=FALSE)
vsd <- vst(dds, blind=FALSE)

data <- plotPCA(vsd, intgroup=c("population", "devstage", "sex"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))

data$devstage <- factor(data$devstage, levels=c("L3L", "PP1", "PD1", "AD4"), labels = c("L3L", "PP1", "PD1", "AD4"))

pdf(file="PCA_sex_by_stage.pdf", height=5.5, width=5.5)
ggplot(data, aes(PC1, PC2, color=sex, shape=devstage))  + 
  geom_point(size=4, alpha=0.85) +
  xlab(paste0("PC1: ",percentVar[1], "% variance")) +
  ylab(paste0("PC2: ",percentVar[2], "% variance")) +
  theme_minimal()  
dev.off()                        

ggplot(data, aes(PC1, PC2, color=population, shape=devstage))  + 
  geom_point(size=4, alpha=0.85) +
  xlab(paste0("PC1: ",percentVar[1], "% variance")) +
  ylab(paste0("PC2: ",percentVar[2], "% variance")) +
  theme_minimal()  






## the next several lines will make a cluster heatmap of all the samples vs all the samples

sampleDists <- dist(t(assay(rld)))

library("RColorBrewer")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(rld$devstage, rld$population, rld$sex, sep="-")
colnames(sampleDistMatrix) <- NULL

colors <- colorRampPalette(rev(brewer.pal(9,"Greens")))(255)

library("pheatmap")

pheatmap(sampleDistMatrix,clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col=colors)

#In general, it's good practice to load all the libraries at the top at first, so you can run chunks of code separately

#################Let's look at individual genes!!

d <- plotCounts(dds, gene = "OTAU017482-RA", intgroup = (c("population","sex","devstage")), returnData=TRUE)
d

p <-ggplot(d, aes(x=devstage, y=count, shape=sex, colour = population)) +
  theme_minimal() + theme(text = element_text(size=20), panel.grid.major=element_line(colour="grey"))
p <- p + geom_point(position = position_jitter(w=0.3,h=0), size=3) +
  scale_x_discrete(limits=c("L3L","PP1","PD1","AD4"))
p

################ Save outputs

write.csv(res_pop, file="DGE_NCvsWA_pop_classDay6.csv", row.names = T, quote=F)


### change pvalue to -log(pvalue) and export as .csv with rownames

neglogpval <- as.matrix(-log(res_pop$pvalue))
head(neglogpval)

res_pop_negpval <- cbind(row.names(res_pop),neglogpval)
head(res_pop_negpval)

colnames(res_pop_negpval)=c("gene","neglogpval")

write.csv(res_pop_negpval, file="DGE_NCvsWA_pop_neglogpval_classDay6.csv", row.names=F,quote=F)

------
<div id='id-section10'/>
### Page 10:
------
<div id='id-section11'/>
### Page 11:
------
<div id='id-section12'/>
### Page 12:
------
<div id='id-section13'/>
### Page 13:
------
<div id='id-section14'/>
### Page 14:
------
<div id='id-section15'/>
### Page 15:
------
<div id='id-section16'/>
### Page 16:
------
<div id='id-section17'/>
### Page 17:
------
<div id='id-section18'/>
### Page 18:
------
<div id='id-section19'/>
### Page 19:
------
<div id='id-section20'/>
### Page 20:
------
<div id='id-section21'/>
### Page 21:
------
<div id='id-section22'/>
### Page 22:
------
<div id='id-section23'/>
### Page 23:
------
<div id='id-section24'/>
### Page 24:
------
<div id='id-section25'/>
### Page 25:
------
<div id='id-section26'/>
### Page 26:
------
<div id='id-section27'/>
### Page 27:
------
<div id='id-section28'/>
### Page 28:
------
<div id='id-section29'/>
### Page 29:
------
<div id='id-section30'/>
### Page 30:
------
<div id='id-section31'/>
### Page 31:
------
<div id='id-section32'/>
### Page 32:
------
<div id='id-section33'/>
### Page 33:
------
<div id='id-section34'/>
### Page 34:
------
<div id='id-section35'/>
### Page 35:
------
<div id='id-section36'/>
### Page 36:
------
<div id='id-section37'/>
### Page 37:
------
<div id='id-section38'/>
### Page 38:
------
<div id='id-section39'/>
### Page 39:
------
<div id='id-section40'/>
### Page 40:
------
<div id='id-section41'/>
### Page 41:
------
<div id='id-section42'/>
### Page 42:
------
<div id='id-section43'/>
### Page 43:
------
<div id='id-section44'/>
### Page 44:
------
<div id='id-section45'/>
### Page 45:
------
<div id='id-section46'/>
### Page 46:
------
<div id='id-section47'/>
### Page 47:
------
<div id='id-section48'/>
### Page 48:
------
<div id='id-section49'/>
### Page 49:
------
<div id='id-section50'/>
### Page 50:
------
<div id='id-section51'/>
### Page 51:
------
<div id='id-section52'/>
### Page 52:
------
<div id='id-section53'/>
### Page 53:
------
<div id='id-section54'/>
### Page 54:
------
<div id='id-section55'/>
### Page 55:
------
<div id='id-section56'/>
### Page 56:
------
<div id='id-section57'/>
### Page 57:
------
<div id='id-section58'/>
### Page 58:
------
<div id='id-section59'/>
### Page 59:
------
<div id='id-section60'/>
### Page 60:

------